diff -Naur linux-2.6.13/arch/ppc/boot/simple/misc.c linux-2.6.13-pcore/arch/ppc/boot/simple/misc.c
--- linux-2.6.13/arch/ppc/boot/simple/misc.c	2008-04-23 11:39:05.000000000 +0200
+++ linux-2.6.13-pcore/arch/ppc/boot/simple/misc.c	2008-04-22 09:50:15.000000000 +0200
@@ -269,11 +269,6 @@
 	return rec;
 }
 
-void __attribute__ ((weak))
-board_isa_init(void)
-{
-}
-
 /* Allow decompress_kernel to be hooked into.  This is the default. */
 void * __attribute__ ((weak))
 load_kernel(unsigned long load_addr, int num_words, unsigned long cksum,
diff -Naur linux-2.6.13/arch/ppc/platforms/pcore.c linux-2.6.13-pcore/arch/ppc/platforms/pcore.c
--- linux-2.6.13/arch/ppc/platforms/pcore.c	2008-04-23 11:39:05.000000000 +0200
+++ linux-2.6.13-pcore/arch/ppc/platforms/pcore.c	2008-04-29 15:40:40.000000000 +0200
@@ -43,7 +43,7 @@
 static int board_type;
 
 static inline int __init
-pcore_6750_map_irq(struct pci_dev *dev, unsigned char idsel, unsigned char pin)
+pcore_3750_map_irq(struct pci_dev *dev, unsigned char idsel, unsigned char pin)
 {
 	static char pci_irq_table[][4] =
 	/*
@@ -51,13 +51,13 @@
 	 *      A       B       C       D
 	 */
 	{
-		{9,	10,	11,	12},	/* IDSEL 24 - DEC 21554 */
-		{10,	0,	0,	0},	/* IDSEL 25 - DEC 21143 */
-		{11,	12,	9,	10},	/* IDSEL 26 - PMC I */
-		{12,	9,	10,	11},	/* IDSEL 27 - PMC II */
-		{0,	0,	0,	0},	/* IDSEL 28 - unused */
-		{0,	0,	9,	0},	/* IDSEL 29 - unused */
-		{0,	0,	0,	0},	/* IDSEL 30 - Winbond */
+                {9,     10,     11,     12},    /* IDSEL 24 - DEC 21554 PCI-to-PCI bridge */
+                {9,     0,      0,      0},     /* IDSEL 25 - DEC 21143 Ethernet controller */
+                {11,    12,     9,      10},    /* IDSEL 26 - PMC I */
+                {12,    9,      10,     11},    /* IDSEL 27 - PMC II */
+                {0,     0,      0,      0},     /* IDSEL 28 - unused */
+                {0,     0,      0,      0},     /* IDSEL 29 - unused */
+                {0,     0,      0,      0},     /* IDSEL 30 - Winbond */
 		};
 	const long min_idsel = 24, max_idsel = 30, irqs_per_slot = 4;
 	return PCI_IRQ_TABLE_LOOKUP;
@@ -134,7 +134,7 @@
 			PCI_VENDOR_ID,
 			&host_bridge);
 	if (host_bridge == MPC10X_BRIDGE_106)
-		board_type = PCORE_TYPE_6750;
+		board_type = PCORE_TYPE_3750;
 	else /* MPC10X_BRIDGE_107 */
 		board_type = PCORE_TYPE_680;
 
@@ -143,8 +143,8 @@
 	ppc_md.pcibios_fixup = pcore_pcibios_fixup;
 	ppc_md.pci_swizzle = common_swizzle;
 
-	if (board_type == PCORE_TYPE_6750)
-		ppc_md.pci_map_irq = pcore_6750_map_irq;
+	if (board_type == PCORE_TYPE_3750)
+		ppc_md.pci_map_irq = pcore_3750_map_irq;
 	else /* PCORE_TYPE_680 */
 		ppc_md.pci_map_irq = pcore_680_map_irq;
 
@@ -159,13 +159,13 @@
 {
 	seq_printf(m, "vendor\t\t: Force Computers\n");
 
-	if (board_type == PCORE_TYPE_6750)
-		seq_printf(m, "machine\t\t: PowerCore 6750\n");
+	if (board_type == PCORE_TYPE_3750)
+		seq_printf(m, "machine\t\t: PowerCore 3750\n");
 	else /* PCORE_TYPE_680 */
 		seq_printf(m, "machine\t\t: PowerCore 680\n");
 
 	seq_printf(m, "L2\t\t: " );
-	if (board_type == PCORE_TYPE_6750)
+	if (board_type == PCORE_TYPE_3750)
 		switch (readb(PCORE_DCCR_REG) & PCORE_DCCR_L2_MASK)
 		{
 			case PCORE_DCCR_L2_0KB:
@@ -230,13 +230,11 @@
 #endif
 
  	printk(KERN_INFO "Force PowerCore ");
-	if (board_type == PCORE_TYPE_6750)
-		printk("6750\n");
+	if (board_type == PCORE_TYPE_3750)
+		printk("3750\n");
 	else
 		printk("680\n");
 	printk(KERN_INFO "Port by MontaVista Software, Inc. (source@mvista.com)\n");
-	_set_L2CR(L2CR_L2E | _get_L2CR());
-
 }
 
 static void
@@ -244,7 +242,18 @@
 {
 	local_irq_disable();
 	/* Hard reset */
-	writeb(0x11, 0xfe000332);
+        // writeb(0x11, 0xfe000332);
+
+        /* Soft reset */
+        __asm__ __volatile__
+        ("\n\
+         lis     3,0xfff0        \n\
+         ori     3,3,0x0100      \n\
+         mtspr   26,3            \n\
+         li      3,0             \n\
+         mtspr   27,3            \n\
+         rfi                     \n\
+        ");
 	while(1);
 }
 
diff -Naur linux-2.6.13/arch/ppc/platforms/pcore.h linux-2.6.13-pcore/arch/ppc/platforms/pcore.h
--- linux-2.6.13/arch/ppc/platforms/pcore.h	2008-04-23 11:39:05.000000000 +0200
+++ linux-2.6.13-pcore/arch/ppc/platforms/pcore.h	2008-04-29 15:41:23.000000000 +0200
@@ -16,7 +16,7 @@
 
 #include <asm/mpc10x.h>
 
-#define PCORE_TYPE_6750			1
+#define PCORE_TYPE_3750			1
 #define PCORE_TYPE_680			2
 
 #define PCORE_NVRAM_AS0			0x73
