diff -Naur net-tools-1.60/arp.c net-tools-1.60-robox-linux/arp.c
--- net-tools-1.60/arp.c	2001-04-08 19:05:05.000000000 +0200
+++ net-tools-1.60-robox-linux/arp.c	2008-04-14 15:57:57.000000000 +0200
@@ -8,7 +8,7 @@
  *              NET-3 Networking Distribution for the LINUX operating
  *              system.
  *
- * Version:     $Id: arp.c,v 1.20 2001/04/08 17:05:05 pb Exp $
+ * Version:     $Id: arp.c,v 1.23 2003/02/08 19:56:25 ecki Exp $
  *
  * Maintainer:  Bernd 'eckes' Eckenfels, <net-tools@lina.inka.de>
  *
@@ -102,7 +102,7 @@
     struct arpreq req;
     struct sockaddr sa;
     int flags = 0;
-    int err;
+    int deleted = 0;
 
     memset((char *) &req, 0, sizeof(req));
 
@@ -190,35 +190,41 @@
 	}
 	usage();
     }
+
+    // if neighter priv nor pub is given, work on both
     if (flags == 0)
 	flags = 3;
 
     strcpy(req.arp_dev, device);
 
-    err = -1;
+    /* unfortuatelly the kernel interface does not allow us to
+       delete private entries anlone, so we need this hack
+       to avoid "not found" errors if we try both. */
+    deleted = 0;
 
     /* Call the kernel. */
     if (flags & 2) {
 	if (opt_v)
 	    fprintf(stderr, "arp: SIOCDARP(nopub)\n");
-	if ((err = ioctl(sockfd, SIOCDARP, &req) < 0)) {
-	    if (errno == ENXIO) {
+	if (ioctl(sockfd, SIOCDARP, &req) < 0) {
+	    if ((errno == ENXIO) || (errno == ENOENT)) {
 		if (flags & 1)
 		    goto nopub;
 		printf(_("No ARP entry for %s\n"), host);
 		return (-1);
 	    }
-	    perror("SIOCDARP(priv)");
+	    perror("SIOCDARP(nopub)");
 	    return (-1);
-	}
+	} else
+	  deleted = 1;
     }
-    if ((flags & 1) && (err)) {
+    if (!deleted && (flags & 1)) {
       nopub:
 	req.arp_flags |= ATF_PUBL;
 	if (opt_v)
 	    fprintf(stderr, "arp: SIOCDARP(pub)\n");
 	if (ioctl(sockfd, SIOCDARP, &req) < 0) {
-	    if (errno == ENXIO) {
+	    if ((errno == ENXIO) || (errno == ENOENT)) {
 		printf(_("No ARP entry for %s\n"), host);
 		return (-1);
 	    }
@@ -463,7 +469,7 @@
 
     if (!(arp_flags & ATF_COM)) {
 	if (arp_flags & ATF_PUBL)
-	    printf("%-8.8s%-20.20s", "*", "*");
+	    printf("%-8.8s%-20.20s", "*", _("<from_interface>"));
 	else
 	    printf("%-8.8s%-20.20s", "", _("(incomplete)"));
     } else {
@@ -486,7 +492,7 @@
 
     if (!(arp_flags & ATF_COM)) {
 	if (arp_flags & ATF_PUBL)
-	    printf("* ");
+	    printf("<from_interface> ");
 	else
 	    printf(_("<incomplete> "));
     } else {
@@ -499,7 +505,7 @@
     if (arp_flags & ATF_PERM)
 	printf("PERM ");
     if (arp_flags & ATF_PUBL)
-	printf("PUP ");
+	printf("PUB ");
 #ifdef HAVE_ATF_MAGIC
     if (arp_flags & ATF_MAGIC)
 	printf("AUTO ");
@@ -613,9 +619,8 @@
 {
     fprintf(stderr, _("Usage:\n  arp [-vn]  [<HW>] [-i <if>] [-a] [<hostname>]             <-Display ARP cache\n"));
     fprintf(stderr, _("  arp [-v]          [-i <if>] -d  <hostname> [pub][nopub]    <-Delete ARP entry\n"));
-    fprintf(stderr, _("  arp [-vnD] [<HW>] [-i <if>] -f  [<filename>]              <-Add entry from file\n"));
+    fprintf(stderr, _("  arp [-vnD] [<HW>] [-i <if>] -f  [<filename>]            <-Add entry from file\n"));
     fprintf(stderr, _("  arp [-v]   [<HW>] [-i <if>] -s  <hostname> <hwaddr> [temp][nopub] <-Add entry\n"));
-    fprintf(stderr, _("  arp [-v]   [<HW>] [-i <if>] -s  <hostname> <hwaddr> [netmask <nm>] pub  <-''-\n"));
     fprintf(stderr, _("  arp [-v]   [<HW>] [-i <if>] -Ds <hostname> <if> [netmask <nm>] pub      <-''-\n\n"));
     
     fprintf(stderr, _("        -a                       display (all) hosts in alternative (BSD) style\n"));
diff -Naur net-tools-1.60/config.in net-tools-1.60-robox-linux/config.in
--- net-tools-1.60/config.in	2000-05-21 16:32:12.000000000 +0200
+++ net-tools-1.60-robox-linux/config.in	2008-04-14 15:57:57.000000000 +0200
@@ -82,6 +82,8 @@
 bool '(Cisco)-HDLC/LAPB support' HAVE_HWHDLCLAPB n
 bool 'IrDA support' HAVE_HWIRDA y
 bool 'Econet hardware support' HAVE_HWEC n
+bool 'Generic EUI-64 hardware support' HAVE_HWEUI64 n
+
 *
 *
 *           Other Features.
diff -Naur net-tools-1.60/debian/changelog net-tools-1.60-robox-linux/debian/changelog
--- net-tools-1.60/debian/changelog	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/debian/changelog	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1,179 @@
+net-tools (1.60-10) unstable; urgency=low
+
+  * typo fix in po/de.po for german arp command output (Closes: Bug #176151)
+  * added diagnostics messages to mii-tool.8 (Closes: Bug #239229)
+  * new version of nstrcmp (Closes: Bug #226503)
+  * enable EUI64 support
+  * stadanrds version 3.6.1 (no changes)
+
+ -- Bernd Eckenfels <ecki@debian.org>  Fri, 23 Apr 2004 00:57:20 +0200
+
+net-tools (1.60-9) unstable; urgency=medium
+
+  * debian/changelog: fixed in last entry fixed bug from 197925 to 197924
+  * include linux/sockios.h for SIOCGMIIPHY definition (new style)
+    which will fix MII reporting on 2.4 kernels. (Closes: Bug #133648)
+    NOTE: this will now require root to work, it will also not work on
+    older kernels.
+  * Rene Engelhard's patch to make AF X25 compile with 2.6 headers (Closes: Bug #223091)
+
+ -- Bernd Eckenfels <ecki@debian.org>  Mon, 29 Dec 2003 13:42:20 +0100
+
+net-tools (1.60-8) unstable; urgency=medium
+
+  * lib/interface.c: backed off change which skipped reading ioctl 
+    interfacelist, if _proc interfacelist was ok, which does not work,
+    because alias interfaces are only in ioctl list available. It is 
+    also falling back to the original lib/nstrcml.c because the
+    comparision if : is present does not work.
+    (Closes: bug #197924, #197582, #197269)
+  * bumped standards version (no changes)
+
+ -- Bernd Eckenfels <ecki@debian.org>  Sat, 28 Jun 2003 02:56:38 +0200
+
+net-tools (1.60-7) unstable; urgency=low
+
+  * interfaces.c: cvs version fixes Connectiva Bug #5711 (Closes: #149576)
+    from Eduardo Pereira Habkost
+  * netstat.c: cvs version adds support for tcp6,udp6,icmp6 protocols
+  * netstat.c: cvs version adds support for -4 or -6 cmd line shortcut
+  * netstat.c: cvs version adds fix for inode 
+    signedness (Closes: #134600, #78932)
+  * ifconfig: removed -i from usage (Closes: #181528)
+  * make gcc 3.3 happy: added ull unsigned long long prefix to constant
+  * lib/ddp_gr.c+pathnames.h: first support for appletalk routing from cvs
+  * lib/eui64.c+hw.c+Makefile: forst support for new hw type from cvs
+  * lib/interface.c: cvs fix for column run-into (Closes: #161080)
+  * lib/irda.c,lib/ipx.c: cvs update
+  * lib/nstrcmp.c: new, faster version, fixes b-lookups for ifconfig
+  * hostanme.c: gcc 3.3 fix from cvs (not compiled in net-tools.deb)
+  * lib/interface: fixed memory globbering
+    (Closes: #135744, #149579, #185187)
+
+ -- Bernd Eckenfels <ecki@debian.org>  Thu, 12 Jun 2003 05:44:38 +0200
+
+net-tools (1.60-6) unstable; urgency=low
+
+  * took route.c from upstream cvs: allow -6 and -4 option
+  * took statistics.c from upstream cvs: more complete netstat -S
+  * took README from upstream cvs: fix COPYING file title and removed
+    the line (Closes: #102139)
+  * took arp.c from upstream: fixed name "PUB", better not found handling,
+    implicite proxy arp mac changed from '*' to '<from_interface>',
+    usage line made shorter
+  * added mii-tool in control file (Closes: #172473)
+  * fixed description of binary multiple in man page (Closes: #182487)
+  * fixed interpunctation in netstat(8) (Closes: #191660)
+  * fixed compile warning in inet_sr.c by removing empty default: label in
+    switch
+
+ -- Bernd Eckenfels <ecki@debian.org>  Wed, 28 May 2003 22:35:12 +0200
+
+net-tools (1.60-5) unstable; urgency=low
+
+  * thanks Andrew for the 4.1 NMU
+  * NMU: Apply nameif patch from Matt Domsch. (Closes: #178209)
+  * Fix FTB bug (string concatenation in gcc 3.3) (Closes: #194995)
+  * Fix segfault in mii-tool by using cvs version (Closes: #139027)
+  * fixed plural of authors to make lintian happy (I JOIN YOU :)
+
+ -- Bernd Eckenfels <ecki@debian.org>  Wed, 28 May 2003 22:04:14 +0200
+
+net-tools (1.60-4.1) unstable; urgency=low
+
+  * NMU
+  * Apply nameif patch from Matt Domsch. (Closes: #178209)
+
+ -- Andrew Suffield <asuffield@debian.org>  Wed,  9 Apr 2003 21:44:09 +0100
+
+net-tools (1.60-4) unstable; urgency=medium
+
+  * fixed (upstream) spurious newline in ifconfig
+    Thanks Jonathen, Closes: Bug #109379
+  * fixed SI-Units printout in ifconfig (Closes: Bug #110629, #97029, #100167)
+  * this is done by including lib/interfaces.c from upstream cvs
+  * also added ifconfig man page from upstream cvs
+  * fixed (upstream) man page about netstat's --interface opt. (Closes: Bug#120475)
+
+ -- Bernd Eckenfels <ecki@debian.org>  Sat, 24 Nov 2001 06:26:37 +0100
+
+net-tools (1.60-3) unstable; urgency=medium
+
+  * fixed upstream error where counters in ifconfig/netstat -i are 0
+
+ -- Bernd Eckenfels <ecki@debian.org>  Sat, 10 Nov 2001 18:12:13 +0100
+
+net-tools (1.60-2) unstable; urgency=low
+
+  * Fixed the following bug: (Closes: #117837)
+  * inserted ifconfig.c from upstream cvs to do this
+  * bumped debian Standard 3.1.1 -> 3.5.6 and removed -g from linking
+
+ -- Bernd Eckenfels <ecki@debian.org>  Thu,  1 Nov 2001 02:51:33 +0100
+
+net-tools (1.60-1) unstable; urgency=low
+
+  * New upstream
+  * Phil fixed the following Bugs upstream: (Closes: #91919, #93048, #90282)
+
+ -- Bernd Eckenfels <ecki@debian.org>  Mon, 16 Apr 2001 02:28:12 +0200
+
+net-tools (1.59-1) unstable; urgency=low
+
+  * added Build Dependencies (closes bug: #89083)
+  * new upstream version
+  * removed dh_suidregister and a few comments from rules file
+
+ -- Bernd Eckenfels <ecki@debian.org>  Sun, 18 Mar 2001 03:00:33 +0100
+
+net-tools (1.58-2) unstable; urgency=low
+
+  * backported ifconfig change from 1.59 to avoid closing socket too early
+    on some systems (depending on the protocols installed). 
+    Closes Bug: #85688, #85743
+
+ -- Bernd Eckenfels <ecki@debian.org>  Tue, 13 Feb 2001 01:39:59 +0100
+
+net-tools (1.58-1) unstable; urgency=low
+
+  * new upstream version
+  * removed local man pages for now (Closes: bug #83894)
+  * install nameif in /sbin
+
+ -- Bernd Eckenfels <ecki@debian.org>  Sat, 10 Feb 2001 21:50:30 +0100
+
+net-tools (1.57-2) unstable; urgency=high
+
+  * rebuild, this is believed to closes: #75825
+
+ -- Bernd Eckenfels <ecki@debian.org>  Sat, 30 Dec 2000 22:43:01 +0100
+
+net-tools (1.57-1) unstable; urgency=low
+
+  * new debian maintainer (thanks Anthony for the Work!)
+  * new upstream version
+
+ -- Bernd Eckenfels <ecki@debian.org>  Mon, 14 Aug 2000 02:40:13 +0200
+
+net-tools (1.54-3) unstable; urgency=low
+
+  * Argggh. Don't use that horrible dh_installmanpages hack. (Closes:
+    Bug#68925, Bug#68879)
+
+ -- Anthony Towns <ajt@debian.org>  Sun, 13 Aug 2000 00:12:05 +1000
+
+net-tools (1.54-2) unstable; urgency=low
+
+  * Reapply slattach patch from netbase 3.16-3.
+
+ -- Anthony Towns <ajt@debian.org>  Thu, 10 Aug 2000 11:32:58 +1000
+
+net-tools (1.54-1) unstable; urgency=low
+
+  * Split from netbase.
+
+ -- Anthony Towns <ajt@debian.org>  Mon, 17 Jul 2000 07:34:12 +1000
+
+Local variables:
+mode: debian-changelog
+End:
diff -Naur net-tools-1.60/debian/config.h net-tools-1.60-robox-linux/debian/config.h
--- net-tools-1.60/debian/config.h	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/debian/config.h	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1,75 @@
+/*
+* config.h	Automatically generated configuration includefile
+*
+* NET-TOOLS	A collection of programs that form the base set of the
+*		NET-3 Networking Distribution for the LINUX operating
+*		system.
+*
+*		DO  NOT  EDIT  DIRECTLY
+*
+*/
+
+/* 
+ * 
+ * Internationalization
+ * 
+ * The net-tools package has currently been translated to French,
+ * German and Brazilian Portugese.  Other translations are, of
+ * course, welcome.  Answer `n' here if you have no support for
+ * internationalization on your system.
+ * 
+ */
+#define I18N 1
+
+/* 
+ * 
+ * Protocol Families.
+ * 
+ */
+#define HAVE_AFUNIX 1
+#define HAVE_AFINET 1
+#define HAVE_AFINET6 1
+#define HAVE_AFIPX 1
+#define HAVE_AFATALK 1
+#define HAVE_AFAX25 1
+#define HAVE_AFNETROM 1
+#define HAVE_AFROSE 1
+#define HAVE_AFX25 1
+#define HAVE_AFECONET 1
+#define HAVE_AFDECnet 1
+#define HAVE_AFASH 1
+
+/* 
+ * 
+ * Device Hardware types.
+ * 
+ */
+#define HAVE_HWETHER 1
+#define HAVE_HWARC 1
+#define HAVE_HWSLIP 1
+#define HAVE_HWPPP 1
+#define HAVE_HWTUNNEL 1
+#define HAVE_HWSTRIP 1
+#define HAVE_HWTR 1
+#define HAVE_HWAX25 1
+#define HAVE_HWROSE 1
+#define HAVE_HWNETROM 1
+#define HAVE_HWX25 1
+#define HAVE_HWFR 1
+#define HAVE_HWSIT 1
+#define HAVE_HWFDDI 1
+#define HAVE_HWHIPPI 1
+#define HAVE_HWASH 1
+#define HAVE_HWHDLCLAPB 1
+#define HAVE_HWIRDA 1
+#define HAVE_HWEC 1
+#define HAVE_HWEUI64 1
+
+/* 
+ * 
+ * Other Features.
+ * 
+ */
+#define HAVE_FW_MASQUERADE 1
+#define HAVE_IP_TOOLS 1
+#define HAVE_MII 1
diff -Naur net-tools-1.60/debian/config.make net-tools-1.60-robox-linux/debian/config.make
--- net-tools-1.60/debian/config.make	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/debian/config.make	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1,36 @@
+I18N=1
+HAVE_AFUNIX=1
+HAVE_AFINET=1
+HAVE_AFINET6=1
+HAVE_AFIPX=1
+HAVE_AFATALK=1
+HAVE_AFAX25=1
+HAVE_AFNETROM=1
+HAVE_AFROSE=1
+HAVE_AFX25=1
+HAVE_AFECONET=1
+HAVE_AFDECnet=1
+HAVE_AFASH=1
+HAVE_HWETHER=1
+HAVE_HWARC=1
+HAVE_HWSLIP=1
+HAVE_HWPPP=1
+HAVE_HWTUNNEL=1
+HAVE_HWSTRIP=1
+HAVE_HWTR=1
+HAVE_HWAX25=1
+HAVE_HWROSE=1
+HAVE_HWNETROM=1
+HAVE_HWX25=1
+HAVE_HWFR=1
+HAVE_HWSIT=1
+HAVE_HWFDDI=1
+HAVE_HWHIPPI=1
+HAVE_HWASH=1
+HAVE_HWHDLCLAPB=1
+HAVE_HWIRDA=1
+HAVE_HWEC=1
+HAVE_EUI64=1
+HAVE_FW_MASQUERADE=1
+HAVE_IP_TOOLS=1
+HAVE_MII=1
diff -Naur net-tools-1.60/debian/control net-tools-1.60-robox-linux/debian/control
--- net-tools-1.60/debian/control	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/debian/control	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1,20 @@
+Source: net-tools
+Section: net
+Priority: important
+Build-Depends: debhelper, gettext
+Maintainer: Bernd Eckenfels <ecki@debian.org>
+Standards-Version: 3.6.1
+
+Package: net-tools
+Architecture: any
+Depends: ${shlibs:Depends}
+Replaces: netbase (<< 4.00)
+Description: The NET-3 networking toolkit
+ This package includes the important tools for controlling the network
+ subsystem of the Linux kernel.  This includes arp, ifconfig, netstat, 
+ rarp, nameif and route.  Additionally, this package contains utilities
+ relating to particular network hardware types (plipconfig, slattach, 
+ mii-tool) and advanced aspects of IP configuration (iptunnel, ipmaddr).
+ .
+ In the upstream package 'hostname' and friends are included. Those are
+ not installed by this package, since there is a special "hostname*.deb".
diff -Naur net-tools-1.60/debian/copyright net-tools-1.60-robox-linux/debian/copyright
--- net-tools-1.60/debian/copyright	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/debian/copyright	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1,18 @@
+This debian package is maintained by Bernd Eckenfels <ecki@debian.org> since 
+Mon, 14 Aug 2000 02:42:13 +0200. Debian Informations will be uploaded
+upstream, too.
+
+This package was debianized by Anthony Towns <ajt@debian.org> on
+Mon, 17 Jul 2000 07:34:12 +1000. It was originally part of the netbase
+package.
+
+It was downloaded from http://www.tazenda.demon.co.uk/phil/net-tools/
+
+Upstream Authors: Phil Blundell <philb@gnu.org>, 
+                  Bernd Eckenfels <net-tools@lina.inka.de>
+
+Copyright:
+
+Distributed under the terms of the GNU General Public License version 2,
+as published by the Free Software Foundation. On Debian systems you can
+find a copy of this license in /usr/share/common-licenses/GPL.
diff -Naur net-tools-1.60/debian/dirs net-tools-1.60-robox-linux/debian/dirs
--- net-tools-1.60/debian/dirs	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/debian/dirs	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1 @@
+usr/sbin
diff -Naur net-tools-1.60/debian/docs net-tools-1.60-robox-linux/debian/docs
--- net-tools-1.60/debian/docs	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/debian/docs	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1,3 @@
+README
+README.ipv6
+TODO
diff -Naur net-tools-1.60/debian/rules net-tools-1.60-robox-linux/debian/rules
--- net-tools-1.60/debian/rules	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/debian/rules	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1,74 @@
+#!/usr/bin/make -f
+# Sample debian/rules that uses debhelper.
+# GNU copyright 1997 to 1999 by Joey Hess.
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+# This is the debhelper compatability version to use.
+export DH_COMPAT=1
+
+configure: configure-stamp
+configure-stamp:
+	dh_testdir
+	touch configure-stamp
+
+build: configure-stamp build-stamp
+build-stamp:
+	dh_testdir
+	cp debian/config.h config.h
+	cp debian/config.make config.make
+	$(MAKE)
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp configure-stamp
+	-$(MAKE) clobber
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+	$(MAKE) update BASEDIR=`pwd`/debian/tmp
+	mv debian/tmp/sbin/arp debian/tmp/usr/sbin/arp
+	# we don't want man pages for domainname and friends... (yet)
+	rm -rf debian/tmp/usr/share/man/*/man1/ debian/tmp/usr/share/man/man1/
+	# we don't install local manpages until policy is clear
+	rm -rf debian/tmp/usr/share/man/*_*/
+
+
+# Build architecture-independent files here.
+binary-indep: build install
+# We have nothing to do by default.
+
+# Build architecture-dependent files here.
+binary-arch: build install
+#	dh_testversion
+	dh_testdir
+	dh_testroot
+#	dh_installdebconf	
+	dh_installdocs
+	dh_installexamples
+	dh_installmenu
+	dh_installcron
+#	dh_installmanpages
+	dh_installinfo
+#	dh_undocumented
+	dh_installchangelogs 
+	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+#	dh_makeshlibs
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol -- -isp
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install
diff -Naur net-tools-1.60/hostname.c net-tools-1.60-robox-linux/hostname.c
--- net-tools-1.60/hostname.c	2001-04-08 19:04:23.000000000 +0200
+++ net-tools-1.60-robox-linux/hostname.c	2008-04-14 15:57:57.000000000 +0200
@@ -32,6 +32,7 @@
  */
 #include <stdio.h>
 #include <unistd.h>
+#include <stdlib.h>
 #include <getopt.h>
 #include <string.h>
 #include <netdb.h>
@@ -97,7 +98,6 @@
 	case EINVAL:
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
-	default:
 	}
 	exit(1);
     };
@@ -116,7 +116,6 @@
 	case EINVAL:
 	    fprintf(stderr, _("%s: name too long\n"), program_name);
 	    break;
-	default:
 	}
 	exit(1);
     };
@@ -173,7 +172,6 @@
 	    *p = '\0';
 	printf("%s\n", hp->h_name);
 	break;
-    default:
     }
 }
 
diff -Naur net-tools-1.60/ifconfig.c net-tools-1.60-robox-linux/ifconfig.c
--- net-tools-1.60/ifconfig.c	2001-04-13 20:25:18.000000000 +0200
+++ net-tools-1.60-robox-linux/ifconfig.c	2008-04-14 15:57:57.000000000 +0200
@@ -3,7 +3,7 @@
  *              that either displays or sets the characteristics of
  *              one or more of the system's networking interfaces.
  *
- * Version:     $Id: ifconfig.c,v 1.50 2001/04/13 18:25:18 pb Exp $
+ * Version:     $Id: ifconfig.c,v 1.53 2001/11/01 01:54:49 ecki Exp $
  *
  * Author:      Fred N. van Kempen, <waltje@uwalt.nl.mugnet.org>
  *              and others.  Copyright 1993 MicroWalt Corporation
@@ -88,7 +88,6 @@
 char *Release = RELEASE, *Version = "ifconfig 1.42 (2001-04-13)";
 
 int opt_a = 0;			/* show all interfaces          */
-int opt_i = 0;			/* show the statistics          */
 int opt_v = 0;			/* debugging output flag        */
 
 int addr_family = 0;		/* currently selected AF        */
@@ -105,7 +104,7 @@
     int res;
 
     if (ife_short)
-	printf(_("Iface   MTU Met    RX-OK RX-ERR RX-DRP RX-OVR    TX-OK TX-ERR TX-DRP TX-OVR Flg\n"));
+	printf(_("Iface   MTU Met   RX-OK RX-ERR RX-DRP RX-OVR   TX-OK TX-ERR TX-DRP TX-OVR Flg\n"));
 
     if (!ifname) {
 	res = for_all_interfaces(do_if_print, &opt_a);
@@ -127,7 +126,7 @@
 
     safe_strncpy(ifr.ifr_name, ifname, IFNAMSIZ);
     if (ioctl(skfd, SIOCGIFFLAGS, &ifr) < 0) {
-	fprintf(stderr, _("%s: unknown interface: %s\n"), 
+	fprintf(stderr, _("%s: ERROR while getting interface flags: %s\n"), 
 		ifname,	strerror(errno));
 	return (-1);
     }
@@ -159,7 +158,7 @@
 
     safe_strncpy(ifr.ifr_name, ifname, IFNAMSIZ);
     if (ioctl(fd, SIOCGIFFLAGS, &ifr) < 0) {
-	fprintf(stderr, _("%s: unknown interface: %s\n"), 
+	fprintf(stderr, _("%s: ERROR while getting interface flags: %s\n"), 
 		ifname, strerror(errno));
 	return -1;
     }
@@ -172,9 +171,35 @@
     return (0);
 }
 
+/** test is a specified flag is set */
+static int test_flag(char *ifname, short flags)
+{
+    struct ifreq ifr;
+    int fd;
+
+    if (strchr(ifname, ':')) {
+        /* This is a v4 alias interface.  Downing it via a socket for
+	   another AF may have bad consequences. */
+        fd = get_socket_for_af(AF_INET);
+	if (fd < 0) {
+	    fprintf(stderr, _("No support for INET on this system.\n"));
+	    return -1;
+	}
+    } else
+        fd = skfd;
+
+    safe_strncpy(ifr.ifr_name, ifname, IFNAMSIZ);
+    if (ioctl(fd, SIOCGIFFLAGS, &ifr) < 0) {
+	fprintf(stderr, _("%s: ERROR while testing interface flags: %s\n"), 
+		ifname, strerror(errno));
+	return -1;
+    }
+    return (ifr.ifr_flags & flags);
+}
+
 static void usage(void)
 {
-    fprintf(stderr, _("Usage:\n  ifconfig [-a] [-i] [-v] [-s] <interface> [[<AF>] <address>]\n"));
+    fprintf(stderr, _("Usage:\n  ifconfig [-a] [-v] [-s] <interface> [[<AF>] <address>]\n"));
 #if HAVE_AFINET
     fprintf(stderr, _("  [add <address>[/<prefixlen>]]\n"));
     fprintf(stderr, _("  [del <address>[/<prefixlen>]]\n"));
@@ -208,7 +233,7 @@
 static void version(void)
 {
     fprintf(stderr, "%s\n%s\n", Release, Version);
-    exit(0);
+    exit(E_USAGE);
 }
 
 static int set_netmask(int skfd, struct ifreq *ifr, struct sockaddr *sa)
@@ -222,7 +247,7 @@
 		strerror(errno));
 	err = 1;
     }
-    return 0;
+    return err;
 }
 
 int main(int argc, char **argv)
@@ -388,6 +413,8 @@
 	}
 	if (!strcmp(*spp, "-promisc")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_PROMISC);
+	    if (test_flag(ifr.ifr_name, IFF_PROMISC) > 0)
+	    	fprintf(stderr, _("Warning: Interface %s still in promisc mode... maybe other application is running?\n"), ifr.ifr_name);
 	    spp++;
 	    continue;
 	}
@@ -398,6 +425,8 @@
 	}
 	if (!strcmp(*spp, "-multicast")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_MULTICAST);
+	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    	fprintf(stderr, _("Warning: Interface %s still in MULTICAST mode.\n"), ifr.ifr_name);
 	    spp++;
 	    continue;
 	}
@@ -408,6 +437,8 @@
 	}
 	if (!strcmp(*spp, "-allmulti")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_ALLMULTI);
+	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    	fprintf(stderr, _("Warning: Interface %s still in ALLMULTI mode.\n"), ifr.ifr_name);
 	    spp++;
 	    continue;
 	}
@@ -430,6 +461,8 @@
 	if (!strcmp(*spp, "-dynamic")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_DYNAMIC);
 	    spp++;
+	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    	fprintf(stderr, _("Warning: Interface %s still in DYNAMIC mode.\n"), ifr.ifr_name);
 	    continue;
 	}
 #endif
@@ -486,6 +519,8 @@
 
 	if (!strcmp(*spp, "-broadcast")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_BROADCAST);
+	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    	fprintf(stderr, _("Warning: Interface %s still in BROADCAST mode.\n"), ifr.ifr_name);
 	    spp++;
 	    continue;
 	}
@@ -541,7 +576,7 @@
 		continue;
 	    }
 	    didnetmask++;
-	    goterr = set_netmask(ap->fd, &ifr, &sa);
+	    goterr |= set_netmask(ap->fd, &ifr, &sa);
 	    spp++;
 	    continue;
 	}
@@ -613,6 +648,8 @@
 	if (!strcmp(*spp, "-pointopoint")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_POINTOPOINT);
 	    spp++;
+	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    	fprintf(stderr, _("Warning: Interface %s still in POINTOPOINT mode.\n"), ifr.ifr_name);
 	    continue;
 	}
 	if (!strcmp(*spp, "pointopoint")) {
@@ -911,7 +948,7 @@
 		if (didnetmask)
 		    usage();
 
-		goterr = set_netmask(skfd, &ifr, &sa);
+		goterr |= set_netmask(skfd, &ifr, &sa);
 		didnetmask++;
 		break;
 	    }
@@ -980,6 +1017,9 @@
 	spp++;
     }
 
+    if (opt_v && goterr)
+    	fprintf(stderr, _("WARNING: at least one error occured. (%d)\n"), goterr);
+
     return (goterr);
 }
 
diff -Naur net-tools-1.60/include/interface.h net-tools-1.60-robox-linux/include/interface.h
--- net-tools-1.60/include/interface.h	2001-02-10 20:24:25.000000000 +0100
+++ net-tools-1.60-robox-linux/include/interface.h	2008-04-14 15:57:57.000000000 +0200
@@ -64,13 +64,17 @@
 extern int if_fetch(struct interface *ife);
 
 extern int for_all_interfaces(int (*)(struct interface *, void *), void *);
-extern int free_interface_list(void);
+extern int if_cache_free(void);
 extern struct interface *lookup_interface(char *name);
 extern int if_readlist(void);
 
 extern int do_if_fetch(struct interface *ife);
 extern int do_if_print(struct interface *ife, void *cookie);
 
+extern int    procnetdev_version(char *buf);
+extern int    get_dev_fields(char *bp, struct interface *ife);
+extern char * get_name(char *name, char *p);
+
 extern void ife_print(struct interface *ptr);
 
 extern int ife_short;
diff -Naur net-tools-1.60/include/mii.h net-tools-1.60-robox-linux/include/mii.h
--- net-tools-1.60/include/mii.h	2000-05-21 16:21:27.000000000 +0200
+++ net-tools-1.60-robox-linux/include/mii.h	2008-04-14 15:57:57.000000000 +0200
@@ -6,11 +6,14 @@
  * Copyright (C) 2000 David A. Hinds -- dhinds@pcmcia.sourceforge.org
  */
 
-#ifndef _LINUX_MII_H
-#define _LINUX_MII_H
+#ifndef _NETTOOL_MII_H
+#define _NETTOOLS_MII_H
+
+#include <linux/sockios.h>
 
 /* network interface ioctl's for MII commands */
 #ifndef SIOCGMIIPHY
+#warning "SIOCGMIIPHY is not defined by your kernel source"
 #define SIOCGMIIPHY (SIOCDEVPRIVATE)	/* Read from current PHY */
 #define SIOCGMIIREG (SIOCDEVPRIVATE+1) 	/* Read any PHY register */
 #define SIOCSMIIREG (SIOCDEVPRIVATE+2) 	/* Write any PHY register */
@@ -83,4 +86,4 @@
 #define  MII_ANER_PAGE_RX	0x0002
 #define  MII_ANER_LP_AN_ABLE	0x0001
 
-#endif /* _LINUX_MII_H */
+#endif /* _NETTOOLS_MII_H */
diff -Naur net-tools-1.60/lib/ddp_gr.c net-tools-1.60-robox-linux/lib/ddp_gr.c
--- net-tools-1.60/lib/ddp_gr.c	1999-01-09 16:54:32.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/ddp_gr.c	2008-04-14 15:57:57.000000000 +0200
@@ -1,3 +1,20 @@
+/*
+ * lib/ddp_gr.c         Prinbting of DDP (AppleTalk) routing table
+ *                      used by the NET-LIB.
+ *
+ * NET-LIB      
+ *
+ * Version:     $Id: ddp_gr.c,v 1.4 2002/06/02 05:25:15 ecki Exp $
+ *
+ * Author:      Ajax <ajax@firest0rm.org>
+ *
+ * Modification:
+ *  2002-06-02 integrated into main source by Bernd Eckenfels
+ *
+ */
+
+/* TODO: name lookups (/etc/atalk.names?  NBP?) */
+
 #include "config.h"
 
 #if HAVE_AFATALK
@@ -16,9 +33,61 @@
 #include "pathnames.h"
 #include "intl.h"
 
+/* stolen from inet_gr.c */
+#define flags_decode(i,o) do {                  \
+        o[0] = '\0';                            \
+        if (i & RTF_UP) strcat(o, "U");         \
+        if (i & RTF_GATEWAY) strcat(o, "G");    \
+        if (i & RTF_REJECT) strcat(o, "!");     \
+        if (i & RTF_HOST) strcat(o, "H");       \
+        if (i & RTF_REINSTATE) strcat(o, "R");  \
+        if (i & RTF_DYNAMIC) strcat(o, "D");    \
+        if (i & RTF_MODIFIED) strcat(o, "M");   \
+        if (i & RTF_DEFAULT) strcat(o, "d");    \
+        if (i & RTF_ALLONLINK) strcat(o, "a");  \
+        if (i & RTF_ADDRCONF) strcat(o, "c");   \
+        if (i & RTF_NONEXTHOP) strcat(o, "o");  \
+        if (i & RTF_EXPIRES) strcat(o, "e");    \
+        if (i & RTF_CACHE) strcat(o, "c");      \
+        if (i & RTF_FLOW) strcat(o, "f");       \
+        if (i & RTF_POLICY) strcat(o, "p");     \
+        if (i & RTF_LOCAL) strcat(o, "l");      \
+        if (i & RTF_MTU) strcat(o, "u");        \
+        if (i & RTF_WINDOW) strcat(o, "w");     \
+        if (i & RTF_IRTT) strcat(o, "i");       \
+        if (i & RTF_NOTCACHED) strcat(o, "n");  \
+    } while (0)
+
 int DDP_rprint(int options)
 {
-    fprintf(stderr, _("Routing table for `ddp' not yet supported.\n"));
-    return (1);
+    FILE *fp;
+    char *dest, *gw, *dev, *flags;
+    char oflags[32];
+    char *hdr = "Destination     Gateway         Device          Flags";
+
+    fp = fopen(_PATH_PROCNET_ATALK_ROUTE, "r");
+
+    if (!fp) {
+        perror("Error opening " _PATH_PROCNET_ATALK_ROUTE);
+        fprintf(stderr, "DDP (AppleTalk) not configured on this system.\n");
+        return 1;
+    }
+
+    fscanf(fp, "%as %as %as %as\n", &dest, &gw, &flags, &dev);
+    free(dest); free(gw); free(dev); free(flags);
+
+    printf("%s\n", hdr);
+
+    while (fscanf(fp, "%as %as %as %as\n", &dest, &gw, &flags, &dev) == 4) {
+        int iflags = atoi(flags);
+        flags_decode(iflags, oflags);
+        printf("%-16s%-16s%-16s%-s\n", dest, gw, dev, oflags);
+        free(dest); free(gw); free(dev); free(flags);
+    }
+
+    fclose(fp);
+    
+    return 0;
+
 }
 #endif
diff -Naur net-tools-1.60/lib/ether.c net-tools-1.60-robox-linux/lib/ether.c
--- net-tools-1.60/lib/ether.c	1999-11-20 22:02:53.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/ether.c	2008-04-14 15:57:57.000000000 +0200
@@ -2,7 +2,7 @@
  * lib/ether.c        This file contains an implementation of the "Ethernet"
  *              support functions.
  *
- * Version:     $Id: ether.c,v 1.7 1999/09/27 11:00:47 philip Exp $
+ * Version:     $Id: ether.c,v 1.8 2002/07/30 05:17:29 ecki Exp $
  *
  * Author:      Fred N. van Kempen, <waltje@uwalt.nl.mugnet.org>
  *              Copyright 1993 MicroWalt Corporation
diff -Naur net-tools-1.60/lib/eui64.c net-tools-1.60-robox-linux/lib/eui64.c
--- net-tools-1.60/lib/eui64.c	1970-01-01 01:00:00.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/eui64.c	2008-04-14 15:57:57.000000000 +0200
@@ -0,0 +1,155 @@
+/*
+ * lib/eui64.c  This file contains support for generic EUI-64 hw addressing
+ *
+ * Version:     $Id: eui64.c,v 1.1 2001/11/12 02:12:05 ecki Exp $
+ *
+ * Author:      Daniel Stodden <stodden@in.tum.de>
+ *              Copyright 2001 Daniel Stodden
+ *
+ *              blueprinted from ether.c 
+ *              Copyright 1993 MicroWalt Corporation
+ *
+ *              This program is free software; you can redistribute it
+ *              and/or  modify it under  the terms of  the GNU General
+ *              Public  License as  published  by  the  Free  Software
+ *              Foundation;  either  version 2 of the License, or  (at
+ *              your option) any later version.
+ */
+#include "config.h"
+
+#if HAVE_HWEUI64
+
+#include <sys/types.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <net/if_arp.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <termios.h>
+#include <unistd.h>
+#include "net-support.h"
+#include "pathnames.h"
+#include "intl.h"
+
+/*
+ * EUI-64 constants
+ */
+
+#define EUI64_ALEN	8
+
+#ifndef ARPHRD_EUI64
+#define ARPHRD_EUI64	27
+#warning "ARPHRD_EUI64 not defined in <net/if_arp.h>. Using private value 27"
+#endif
+
+struct hwtype eui64_hwtype;
+
+/* Display an EUI-64 address in readable format. */
+static char *pr_eui64( unsigned char *ptr )
+{
+	static char buff[64];
+
+	snprintf(buff, sizeof(buff), "%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X",
+		 (ptr[0] & 0377), (ptr[1] & 0377), (ptr[2] & 0377), (ptr[3] & 0377), 
+		 (ptr[4] & 0377), (ptr[5] & 0377), (ptr[6] & 0377), (ptr[7] & 0377)
+		);
+	return (buff);
+}
+
+/* Start the PPP encapsulation on the file descriptor. */
+static int in_eui64( char *bufp, struct sockaddr *sap )
+{
+	unsigned char *ptr;
+	char c, *orig;
+	int i;
+	unsigned val;
+
+	sap->sa_family = eui64_hwtype.type;
+	ptr = sap->sa_data;
+
+	i = 0;
+	orig = bufp;
+	
+	while ((*bufp != '\0') && (i < EUI64_ALEN)) {
+		val = 0;
+		c = *bufp++;
+		if (isdigit(c))
+			val = c - '0';
+		else if (c >= 'a' && c <= 'f')
+			val = c - 'a' + 10;
+		else if (c >= 'A' && c <= 'F')
+			val = c - 'A' + 10;
+		else {
+#ifdef DEBUG
+			fprintf( stderr, _("in_eui64(%s): invalid eui64 address!\n"), 
+				 orig );
+#endif
+			errno = EINVAL;
+			return (-1);
+		}
+
+		val <<= 4;
+		c = *bufp;
+		if (isdigit(c))
+			val |= c - '0';
+		else if (c >= 'a' && c <= 'f')
+			val |= c - 'a' + 10;
+		else if (c >= 'A' && c <= 'F')
+			val |= c - 'A' + 10;
+		else if (c == ':' || c == 0)
+			val >>= 4;
+		else {
+#ifdef DEBUG
+			fprintf( stderr, _("in_eui64(%s): invalid eui64 address!\n"), 
+				 orig );
+#endif
+			errno = EINVAL;
+			return (-1);
+		}
+
+		if (c != 0)
+			bufp++;
+
+		*ptr++ = (unsigned char) (val & 0377);
+		i++;
+		
+		/* We might get a semicolon here - not required. */
+		if (*bufp == ':') {
+			if (i == EUI64_ALEN) {
+#ifdef DEBUG
+				fprintf(stderr, _("in_eui64(%s): trailing : ignored!\n"),
+					orig)
+#endif
+					; /* nothing */
+			}
+			bufp++;
+		}
+	}
+
+	/* That's it.  Any trailing junk? */
+	if ((i == EUI64_ALEN) && (*bufp != '\0')) {
+#ifdef DEBUG
+		fprintf(stderr, _("in_eui64(%s): trailing junk!\n"), orig);
+		errno = EINVAL;
+		return (-1);
+#endif
+	}
+#ifdef DEBUG
+	fprintf(stderr, "in_eui64(%s): %s\n", orig, pr_eui64(sap->sa_data));
+#endif
+
+    return (0);
+}
+
+struct hwtype eui64_hwtype =
+{
+	"eui64", NULL, /*"EUI-64 addressing", */ ARPHRD_EUI64, EUI64_ALEN,
+	pr_eui64, in_eui64, NULL, 0
+};
+
+
+#endif				/* HAVE_EUI64 */
diff -Naur net-tools-1.60/lib/hw.c net-tools-1.60-robox-linux/lib/hw.c
--- net-tools-1.60/lib/hw.c	2000-05-20 20:27:25.000000000 +0200
+++ net-tools-1.60-robox-linux/lib/hw.c	2008-04-14 15:57:57.000000000 +0200
@@ -2,7 +2,7 @@
  * lib/hw.c   This file contains the top-level part of the hardware
  *              support functions module.
  *
- * Version:     $Id: hw.c,v 1.17 2000/05/20 13:38:10 pb Exp $
+ * Version:     $Id: hw.c,v 1.18 2001/11/12 02:12:05 ecki Exp $
  *
  * Maintainer:  Bernd 'eckes' Eckenfels, <net-tools@lina.inka.de>
  *
@@ -73,6 +73,8 @@
 
 extern struct hwtype ec_hwtype;
 
+extern struct hwtype eui64_hwtype;
+
 static struct hwtype *hwtypes[] =
 {
 
@@ -144,6 +146,9 @@
 #if HAVE_HWX25
     &x25_hwtype,
 #endif
+#if HAVE_HWEUI64
+    &eui64_hwtype,
+#endif
     &unspec_hwtype,
     NULL
 };
@@ -217,6 +222,9 @@
 #if HAVE_HWEC
     ec_hwtype.title = _("Econet");
 #endif
+#if HAVE_HWEUI64
+    eui64_hwtype.title = _("Generic EUI-64");
+#endif
     sVhwinit = 1;
 }
 
diff -Naur net-tools-1.60/lib/inet_sr.c net-tools-1.60-robox-linux/lib/inet_sr.c
--- net-tools-1.60/lib/inet_sr.c	2000-02-20 22:46:45.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/inet_sr.c	2008-04-14 15:57:57.000000000 +0200
@@ -104,7 +104,6 @@
        isnet = 1; break;
     case 2:
        isnet = 0; break;
-    default:
     }
 
     /* Fill in the other fields. */
diff -Naur net-tools-1.60/lib/interface.c net-tools-1.60-robox-linux/lib/interface.c
--- net-tools-1.60/lib/interface.c	2001-02-10 20:31:15.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/interface.c	2008-04-14 15:57:57.000000000 +0200
@@ -7,7 +7,7 @@
    8/2000  Andi Kleen make the list operations a bit more efficient.
    People are crazy enough to use thousands of aliases now.
 
-   $Id: interface.c,v 1.14 2001/02/10 19:31:15 pb Exp $
+   $Id: interface.c,v 1.28 2003/05/29 02:09:14 ecki Exp $
  */
 
 #include "config.h"
@@ -23,6 +23,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <ctype.h>
+#include <string.h>
 
 #if HAVE_AFIPX
 #if (__GLIBC__ > 2) || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 1)
@@ -56,8 +57,8 @@
 {
   /* Keep in step with <linux/netdevice.h> */
     {"unknown", NULL, NULL, NULL},
-    {"10base2", "bnc", "coax", NULL},
     {"10baseT", "utp", "tpe", NULL},
+    {"10base2", "bnc", "coax", NULL},
     {"AUI", "thick", "db15", NULL},
     {"100baseT", NULL, NULL, NULL},
     {"100baseTX", NULL, NULL, NULL},
@@ -87,14 +88,20 @@
 
 int ife_short;
 
+int if_list_all = 0;	/* do we have requested the complete proc list, yet? */
+
 static struct interface *int_list, *int_last;
 
 static int if_readlist_proc(char *);
 
-static struct interface *add_interface(char *name)
+static struct interface *if_cache_add(char *name)
 {
     struct interface *ife, **nextp, *new;
 
+    if (!int_list)
+    	int_last = NULL;
+
+    /* the cache is sorted, so if we hit a smaller if, exit */
     for (ife = int_last; ife; ife = ife->prev) {
 	    int n = nstrcmp(ife->name, name); 
 	    if (n == 0) 
@@ -104,7 +111,7 @@
     }
     new(new); 
     safe_strncpy(new->name, name, IFNAMSIZ); 
-    nextp = ife ? &ife->next : &int_list;
+    nextp = ife ? &ife->next : &int_list; // keep sorting
     new->prev = ife;
     new->next = *nextp; 
     if (new->next) 
@@ -117,19 +124,22 @@
 
 struct interface *lookup_interface(char *name)
 {
-    struct interface *ife = NULL;
-
-    if (if_readlist_proc(name) < 0) 
-	    return NULL; 
-    ife = add_interface(name); 
-    return ife;
+   /* if we have read all, use it */
+   if (if_list_all)
+   	return if_cache_add(name);
+
+   /* otherwise we read a limited list */
+   if (if_readlist_proc(name) < 0)
+   	return NULL;
+ 
+   return if_cache_add(name);
 }
 
 int for_all_interfaces(int (*doit) (struct interface *, void *), void *cookie)
 {
     struct interface *ife;
 
-    if (!int_list && (if_readlist() < 0))
+    if (!if_list_all && (if_readlist() < 0))
 	return -1;
     for (ife = int_list; ife; ife = ife->next) {
 	int err = doit(ife, cookie);
@@ -139,13 +149,15 @@
     return 0;
 }
 
-int free_interface_list(void)
+int if_cache_free(void)
 {
     struct interface *ife;
     while ((ife = int_list) != NULL) {
 	int_list = ife->next;
 	free(ife);
     }
+    int_last = NULL;
+    if_list_all = 0;
     return 0;
 }
 
@@ -180,7 +192,7 @@
 	}
 	if (ifc.ifc_len == sizeof(struct ifreq) * numreqs) {
 	    /* assume it overflowed and try again */
-	    numreqs += 10;
+	    numreqs *= 2;
 	    continue;
 	}
 	break;
@@ -188,7 +200,7 @@
 
     ifr = ifc.ifc_req;
     for (n = 0; n < ifc.ifc_len; n += sizeof(struct ifreq)) {
-	add_interface(ifr->ifr_name);
+	if_cache_add(ifr->ifr_name);
 	ifr++;
     }
     err = 0;
@@ -198,7 +210,7 @@
     return err;
 }
 
-static char *get_name(char *name, char *p)
+char *get_name(char *name, char *p)
 {
     while (isspace(*p))
 	p++;
@@ -206,16 +218,19 @@
 	if (isspace(*p))
 	    break;
 	if (*p == ':') {	/* could be an alias */
-	    char *dot = p, *dotname = name;
-	    *name++ = *p++;
-	    while (isdigit(*p))
-		*name++ = *p++;
-	    if (*p != ':') {	/* it wasn't, backup */
-		p = dot;
-		name = dotname;
+		char *dot = p++;
+ 		while (*p && isdigit(*p)) p++;
+		if (*p == ':') {
+			/* Yes it is, backup and copy it. */
+			p = dot;
+			*name++ = *p++;
+			while (*p && isdigit(*p)) {
+				*name++ = *p++;
+			}
+		} else {
+			/* No, it isn't */
+			p = dot;
 	    }
-	    if (*p == '\0')
-		return NULL;
 	    p++;
 	    break;
 	}
@@ -225,7 +240,7 @@
     return p;
 }
 
-static int procnetdev_version(char *buf)
+int procnetdev_version(char *buf)
 {
     if (strstr(buf, "compressed"))
 	return 3;
@@ -234,12 +249,12 @@
     return 1;
 }
 
-static int get_dev_fields(char *bp, struct interface *ife)
+int get_dev_fields(char *bp, struct interface *ife)
 {
     switch (procnetdev_vsn) {
     case 3:
 	sscanf(bp,
-	"%llu %llu %lu %lu %lu %lu %lu %lu %llu %llu %lu %lu %lu %lu %lu %lu",
+	"%Lu %Lu %lu %lu %lu %lu %lu %lu %Lu %Lu %lu %lu %lu %lu %lu %lu",
 	       &ife->stats.rx_bytes,
 	       &ife->stats.rx_packets,
 	       &ife->stats.rx_errors,
@@ -259,7 +274,7 @@
 	       &ife->stats.tx_compressed);
 	break;
     case 2:
-	sscanf(bp, "%llu %llu %lu %lu %lu %lu %llu %llu %lu %lu %lu %lu %lu",
+	sscanf(bp, "%Lu %Lu %lu %lu %lu %lu %Lu %Lu %lu %lu %lu %lu %lu",
 	       &ife->stats.rx_bytes,
 	       &ife->stats.rx_packets,
 	       &ife->stats.rx_errors,
@@ -277,7 +292,7 @@
 	ife->stats.rx_multicast = 0;
 	break;
     case 1:
-	sscanf(bp, "%llu %lu %lu %lu %lu %llu %lu %lu %lu %lu %lu",
+	sscanf(bp, "%Lu %lu %lu %lu %lu %Lu %lu %lu %lu %lu %lu",
 	       &ife->stats.rx_packets,
 	       &ife->stats.rx_errors,
 	       &ife->stats.rx_dropped,
@@ -300,22 +315,16 @@
 
 static int if_readlist_proc(char *target)
 {
-    static int proc_read; 
     FILE *fh;
     char buf[512];
     struct interface *ife;
     int err;
 
-    if (proc_read) 
-	    return 0; 
-    if (!target) 
-	    proc_read = 1;
-
     fh = fopen(_PATH_PROCNET_DEV, "r");
     if (!fh) {
 		fprintf(stderr, _("Warning: cannot open %s (%s). Limited output.\n"),
 			_PATH_PROCNET_DEV, strerror(errno)); 
-		return if_readconf();
+		return -2;
 	}	
     fgets(buf, sizeof buf, fh);	/* eat line */
     fgets(buf, sizeof buf, fh);
@@ -350,7 +359,7 @@
     while (fgets(buf, sizeof buf, fh)) {
 	char *s, name[IFNAMSIZ];
 	s = get_name(name, buf);    
-	ife = add_interface(name);
+	ife = if_cache_add(name);
 	get_dev_fields(s, ife);
 	ife->statistics_valid = 1;
 	if (target && !strcmp(target,name))
@@ -359,7 +368,6 @@
     if (ferror(fh)) {
 	perror(_PATH_PROCNET_DEV);
 	err = -1;
-	proc_read = 0; 
     }
 
 #if 0
@@ -371,9 +379,16 @@
 
 int if_readlist(void) 
 { 
-    int err = if_readlist_proc(NULL); 
-    if (!err)
-	    err = if_readconf();
+    /* caller will/should check not to call this too often 
+     *   (i.e. only if if_list_all == 0 
+     */
+    int err = 0;
+    
+    err |= if_readlist_proc(NULL); 
+    err |= if_readconf();
+
+    if_list_all = 1;
+
     return err;
 } 
 
@@ -580,7 +595,7 @@
 void ife_print_short(struct interface *ptr)
 {
     printf("%-5.5s ", ptr->name);
-    printf("%5d %3d", ptr->mtu, ptr->metric);
+    printf("%5d %-2d ", ptr->mtu, ptr->metric);
     /* If needed, display the interface statistics. */
     if (ptr->statistics_valid) {
 	printf("%8llu %6lu %6lu %6lu",
@@ -636,8 +651,8 @@
     int hf;
     int can_compress = 0;
     unsigned long long rx, tx, short_rx, short_tx;
-    char Rext[5]="b";
-    char Text[5]="b";
+    const char *Rext = "b";
+    const char *Text = "b";
 
 #if HAVE_AFIPX
     static struct aftype *ipxtype = NULL;
@@ -843,10 +858,38 @@
 	tx = ptr->stats.tx_bytes;
 	short_rx = rx * 10;  
 	short_tx = tx * 10;
-	if (rx > 1048576) { short_rx /= 1048576;  strcpy(Rext, "Mb"); }
-	else if (rx > 1024) { short_rx /= 1024;  strcpy(Rext, "Kb"); }
-	if (tx > 1048576) { short_tx /= 1048576;  strcpy(Text, "Mb"); }
-	else if (tx > 1024) { short_tx /= 1024;  strcpy(Text, "Kb"); }
+	if (rx > 1125899906842624ull) {
+	    short_rx /= 1125899906842624ull;
+	    Rext = "PiB";
+	} else if (rx > 1099511627776ull) {
+	    short_rx /= 1099511627776ull;
+	    Rext = "TiB";
+	} else if (rx > 1073741824ull) {
+	    short_rx /= 1073741824ull;
+	    Rext = "GiB";
+	} else if (rx > 1048576) {
+	    short_rx /= 1048576;
+	    Rext = "MiB";
+	} else if (rx > 1024) {
+	    short_rx /= 1024;
+	    Rext = "KiB";
+	}
+	if (tx > 1125899906842624ull) {
+	    short_tx /= 1125899906842624ull;
+	    Text = "PiB";
+	} else 	if (tx > 1099511627776ull) {
+	    short_tx /= 1099511627776ull;
+	    Text = "TiB";
+	} else if (tx > 1073741824ull) {
+	    short_tx /= 1073741824ull;
+	    Text = "GiB";
+	} else if (tx > 1048576) {
+	    short_tx /= 1048576;
+	    Text = "MiB";
+	} else if (tx > 1024) {
+	    short_tx /= 1024;
+	    Text = "KiB";
+	}
 
 	printf("          ");
 	printf(_("TX packets:%llu errors:%lu dropped:%lu overruns:%lu carrier:%lu\n"),
@@ -867,7 +910,7 @@
     }
 
     if ((ptr->map.irq || ptr->map.mem_start || ptr->map.dma ||
-	 ptr->map.base_addr)) {
+	 ptr->map.base_addr >= 0x100)) {
 	printf("          ");
 	if (ptr->map.irq)
 	    printf(_("Interrupt:%d "), ptr->map.irq);
diff -Naur net-tools-1.60/lib/ipx.c net-tools-1.60-robox-linux/lib/ipx.c
--- net-tools-1.60/lib/ipx.c	1999-01-09 16:55:11.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/ipx.c	2008-04-14 15:57:57.000000000 +0200
@@ -133,6 +133,9 @@
     char *ep;
     int nbo;
 
+    if (!sai)
+    	return (-1);
+    	
     sai->sipx_family = AF_IPX;
     sai->sipx_network = htonl(0);
     sai->sipx_node[0] = sai->sipx_node[1] = sai->sipx_node[2] =
diff -Naur net-tools-1.60/lib/irda.c net-tools-1.60-robox-linux/lib/irda.c
--- net-tools-1.60/lib/irda.c	2000-03-05 16:26:04.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/irda.c	2008-04-14 15:57:57.000000000 +0200
@@ -1,13 +1,15 @@
 /*********************************************************************
  *                
  * Filename:      irda.c
- * Version:       0.1
- * Description:   A first attempt to make ifconfig understand IrDA
+ * Version:       0.2
+ * Description:   A second attempt to make ifconfig understand IrDA
  * Status:        Experimental.
  * Author:        Dag Brattli <dagb@cs.uit.no>
  * Created at:    Wed Apr 21 09:03:09 1999
  * Modified at:   Wed Apr 21 09:17:05 1999
  * Modified by:   Dag Brattli <dagb@cs.uit.no>
+ * Modified at:   Wed May  1 11:51:44 CEST 2002
+ * Modified by:   Christoph Bartelmus <christoph@bartelmus.de>
  * 
  *     This program is free software; you can redistribute it and/or 
  *     modify it under the terms of the GNU General Public License as 
@@ -59,9 +61,9 @@
  */
 static char *irda_print(unsigned char *ptr)
 {
-    static char buff[8];
+    static char buff[12];
 
-    sprintf(&buff[strlen(buff)], "%02x:%02x:%02x:%02x", ptr[3], ptr[2], 
+    snprintf(buff, 12, "%02x:%02x:%02x:%02x", ptr[3], ptr[2], 
 	    ptr[1], ptr[0]);
 
     return (buff);
diff -Naur net-tools-1.60/lib/Makefile net-tools-1.60-robox-linux/lib/Makefile
--- net-tools-1.60/lib/Makefile	2000-10-28 12:59:42.000000000 +0200
+++ net-tools-1.60-robox-linux/lib/Makefile	2008-04-14 15:57:57.000000000 +0200
@@ -16,7 +16,7 @@
 #
 
 
-HWOBJS	 = hw.o loopback.o slip.o ether.o ax25.o ppp.o arcnet.o tr.o tunnel.o frame.o sit.o rose.o ash.o fddi.o hippi.o hdlclapb.o strip.o irda.o ec_hw.o x25.o
+HWOBJS	 = hw.o loopback.o slip.o ether.o ax25.o ppp.o arcnet.o tr.o tunnel.o frame.o sit.o rose.o ash.o fddi.o hippi.o hdlclapb.o strip.o irda.o ec_hw.o x25.o eui64.o
 AFOBJS	 = unix.o inet.o inet6.o ax25.o ipx.o ddp.o ipx.o netrom.o af.o rose.o econet.o x25.o
 AFGROBJS = inet_gr.o inet6_gr.o ipx_gr.o ddp_gr.o netrom_gr.o ax25_gr.o rose_gr.o getroute.o x25_gr.o
 AFSROBJS = inet_sr.o inet6_sr.o netrom_sr.o ipx_sr.o setroute.o x25_sr.o
diff -Naur net-tools-1.60/lib/net-features.h net-tools-1.60-robox-linux/lib/net-features.h
--- net-tools-1.60/lib/net-features.h	2000-05-20 20:27:25.000000000 +0200
+++ net-tools-1.60-robox-linux/lib/net-features.h	2008-04-14 15:57:57.000000000 +0200
@@ -295,6 +295,13 @@
 "-"
 #endif
 "HDLC/LAPB "
+
+#if HAVE_HWEUI64
+"+"
+#else
+"-"
+#endif
+"EUI64 "
 ;
 
 
diff -Naur net-tools-1.60/lib/net-support.h net-tools-1.60-robox-linux/lib/net-support.h
--- net-tools-1.60/lib/net-support.h	2000-10-28 12:59:42.000000000 +0200
+++ net-tools-1.60-robox-linux/lib/net-support.h	2008-04-14 15:57:57.000000000 +0200
@@ -15,15 +15,6 @@
  * Author:      Fred N. van Kempen, <waltje@uwalt.nl.mugnet.org>
  *              Copyright 1993 MicroWalt Corporation
  *
- * Modifications:
- *960125 {1.20} Bernd Eckenfels:        reformated, layout
- *960202 {1.30} Bernd Eckenfels:        rprint in aftype
- *960206 {1.31} Bernd Eckenfels:        route_init
- *960219 {1.32} Bernd Eckenfels:        type for ap->input()
- *960322 {1.33} Bernd Eckenfels:        activate_ld and const in get_hwtype
- *960413 {1.34} Bernd Eckenfels:        new RTACTION suport
- *990101 {1.35} Bernd Eckenfels:	print_(hw|af)list support, added kerneldefines
- *
  *              This program is free software; you can redistribute it
  *              and/or  modify it under  the terms of  the GNU General
  *              Public  License as  published  by  the  Free  Software
diff -Naur net-tools-1.60/lib/nstrcmp.c net-tools-1.60-robox-linux/lib/nstrcmp.c
--- net-tools-1.60/lib/nstrcmp.c	1999-01-09 16:55:20.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/nstrcmp.c	2008-04-14 15:57:57.000000000 +0200
@@ -1,34 +1,157 @@
 /* Copyright 1998 by Andi Kleen. Subject to the GPL. */
+/* rewritten by bernd eckenfels because of complicated alias semantic */
 /* $Id: nstrcmp.c,v 1.2 1998/11/15 20:11:38 freitag Exp $ */ 
 #include <ctype.h>
 #include <stdlib.h>
+#include <string.h>
 #include "util.h"
 
-/* like strcmp(), but knows about numbers */
-int nstrcmp(const char *astr, const char *b)
+
+/* return numerical :999 suffix or null. sideeffect: replace ':' with \0 */
+char* cutalias(char* name)
 {
-    const char *a = astr;
+	int digit = 0;
+	int pos;
+	
+	for(pos=strlen(name); pos>0; pos--)
+	{
+		if (name[pos-1]==':' && digit)
+		{
+			name[pos-1]='\0';
+			return name+pos;
+		}
+		if (!isdigit(name[pos-1]))
+			break;
+		digit = 1;
+	}
+	return NULL;
+}
 
-    while (*a == *b) {
-	if (*a == '\0')
-	    return 0;
-	a++;
-	b++;
-    }
-    if (isdigit(*a)) {
-	if (!isdigit(*b))
-	    return -1;
-	while (a > astr) {
-	    a--;
-	    if (!isdigit(*a)) {
-		a++;
-		break;
-	    }
-	    if (!isdigit(*b))
-		return -1;
-	    b--;
+
+/* return index of last non digit or -1 if it does not end with digits */
+int rindex_nondigit(char *name)
+{
+	int pos = strlen(name);
+
+	for(pos=strlen(name); pos>0; pos--)
+	{
+		if (!isdigit(name[pos-1]))
+			return pos;
 	}
-	return atoi(a) > atoi(b) ? 1 : -1;
-    }
-    return *a - *b;
+	return 0;
+}
+
+
+/* like strcmp(), but knows about numbers and ':' alias suffix */
+int nstrcmp(const char *ap, const char *bp)
+{
+	char *a = (char*)strdup(ap);
+	char *b = (char*)strdup(bp);
+	char *an, *bn;
+	int av = 0, bv = 0;
+	char *aalias=cutalias(a);
+	char *balias=cutalias(b);
+	int aindex=rindex_nondigit(a);
+	int bindex=rindex_nondigit(b);
+	int complen=(aindex<bindex)?aindex:bindex;
+	int res = strncmp(a, b, complen);
+
+	if (res != 0)
+		{ free(a); free(b); return res; }
+		
+	if (aindex > bindex)
+		{ free(a); free(b); return 1; }
+		
+	if (aindex < bindex)
+		{ free(a); free(b); return -1; }
+		
+	an = a+aindex;
+	bn = b+bindex;
+	
+	av = atoi(an);
+	bv = atoi(bn);
+	
+	if (av < bv)
+		{ free(a); free(b); return -1; }
+		
+	if (av > bv)
+		{ free(a); free(b); return 1; }
+		
+	av = -1;
+	if (aalias != NULL)
+		av = atoi(aalias);
+	
+	bv = -1;
+	if (balias != NULL)
+		bv = atoi(balias);
+	
+	free(a); free(b);
+	
+	if (av < bv)
+		return -1;
+		
+	if (av > bv)
+		return 1;
+		
+	return 0;
+}
+
+
+#ifdef NSTRCMP_TEST
+
+int cs(int s)
+{
+	if (s < 0) return -1;
+	if (s > 0) return 1;
+	return 0;
+}
+
+
+int dotest(char* a, char* b, int exp)
+{
+	int res = nstrcmp(a, b);
+	int err = (cs(res) != cs(exp));
+	printf("nstrcmp(\"%s\", \"%s\")=%d %d %s\n", a, b, res, exp, err?"WRONG":"OK");
+	return err;
 }
+
+int main()
+{
+	int err = 0;
+	
+	err |= dotest("eth1", "eth1", 0);
+	err |= dotest("eth0:1", "eth0:1", 0);
+	err |= dotest("lan", "lan", 0);
+	err |= dotest("100", "100", 0);
+	err |= dotest("", "", 0);
+	err |= dotest(":", ":", 0);
+	err |= dotest("a:b:c", "a:b:c", 0);
+	err |= dotest("a:", "a:", 0);
+	err |= dotest(":a", ":a", 0);
+
+	err |= dotest("a", "aa", -1);	
+	err |= dotest("eth0", "eth1", -1);
+	err |= dotest("eth1", "eth20", -1);
+	err |= dotest("eth20", "eth100", -1);
+	err |= dotest("eth1", "eth13", -1);
+	err |= dotest("eth", "eth2", -1);
+	err |= dotest("eth0:1", "eth0:2", -1);
+	err |= dotest("eth1:10", "eth13:10", -1);
+	err |= dotest("eth1:1", "eth1:13", -1);
+	err |= dotest("a", "a:", -1);
+	
+	err |= dotest("aa", "a", 1);
+	err |= dotest("eth2", "eth1", 1);
+	err |= dotest("eth13", "eth1", 1);	
+	err |= dotest("eth2", "eth", 1);
+	err |= dotest("eth2:10", "eth2:1", 1);
+	err |= dotest("eth2:5", "eth2:4", 1);
+	err |= dotest("eth3:2", "eth2:3", 1);
+	err |= dotest("eth13:1", "eth1:0", 1);
+	err |= dotest("a:", "a", 1);
+	err |= dotest("a1b12", "a1b2", 1);
+
+	return err;
+}
+
+#endif
diff -Naur net-tools-1.60/lib/pathnames.h net-tools-1.60-robox-linux/lib/pathnames.h
--- net-tools-1.60/lib/pathnames.h	2000-05-20 20:27:26.000000000 +0200
+++ net-tools-1.60-robox-linux/lib/pathnames.h	2008-04-14 15:57:57.000000000 +0200
@@ -1,4 +1,3 @@
-
 /*
  * lib/pathnames.h    This file contains the definitions of the path 
  *                      names used by the NET-LIB.
@@ -45,6 +44,7 @@
 #define _PATH_PROCNET_X25              "/proc/net/x25"
 #define _PATH_PROCNET_X25_ROUTE                "/proc/net/x25_routes"
 #define _PATH_PROCNET_DEV_MCAST		"/proc/net/dev_mcast"
+#define _PATH_PROCNET_ATALK_ROUTE	"/proc/net/atalk_route"
 
 /* pathname for the netlink device */
 #define _PATH_DEV_ROUTE	"/dev/route"
diff -Naur net-tools-1.60/lib/util.h net-tools-1.60-robox-linux/lib/util.h
--- net-tools-1.60/lib/util.h	1999-01-09 16:55:29.000000000 +0100
+++ net-tools-1.60-robox-linux/lib/util.h	2008-04-14 15:57:57.000000000 +0200
@@ -14,3 +14,6 @@
 
 char *safe_strncpy(char *dst, const char *src, size_t size); 
 
+
+#define netmin(a,b) ((a)<(b) ? (a) : (b))
+#define netmax(a,b) ((a)>(b) ? (a) : (b))
diff -Naur net-tools-1.60/lib/x25_sr.c net-tools-1.60-robox-linux/lib/x25_sr.c
--- net-tools-1.60/lib/x25_sr.c	2000-05-20 15:38:10.000000000 +0200
+++ net-tools-1.60-robox-linux/lib/x25_sr.c	2008-04-14 15:57:57.000000000 +0200
@@ -77,7 +77,7 @@
   rt.sigdigits=sigdigits;
 
   /* x25_route_struct.address isn't type struct sockaddr_x25, Why? */
-  memcpy(&rt.address, &sx25.sx25_addr, sizeof(x25_address));
+  memcpy(&rt.address, &sx25.sx25_addr, sizeof(struct x25_address));
 
   while (*args) {
 	if (!strcmp(*args,"device") || !strcmp(*args,"dev")) {
diff -Naur net-tools-1.60/Makefile net-tools-1.60-robox-linux/Makefile
--- net-tools-1.60/Makefile	2001-04-15 16:34:31.000000000 +0200
+++ net-tools-1.60-robox-linux/Makefile	2008-04-14 16:00:27.000000000 +0200
@@ -76,7 +76,7 @@
 NET_LIB_PATH = lib
 NET_LIB_NAME = net-tools
 
-PROGS	:= ifconfig hostname arp netstat route rarp slattach plipconfig nameif
+PROGS	:= ifconfig arp netstat route rarp slattach plipconfig nameif # hostname
 
 -include config.make
 ifeq ($(HAVE_IP_TOOLS),1)
@@ -88,7 +88,7 @@
 
 # Compiler and Linker Options
 # You may need to uncomment and edit these if you are using libc5 and IPv6.
-COPTS = -D_GNU_SOURCE -O2 -Wall -g # -I/usr/inet6/include
+COPTS = -D_GNU_SOURCE -O2 -Wall # -g -I/usr/inet6/include
 ifeq ($(origin LOPTS), undefined)
 LOPTS = 
 endif
@@ -116,6 +116,18 @@
 CFLAGS	= $(COPTS) -I. -idirafter ./include/ -I$(NET_LIB_PATH)
 LDFLAGS	= $(LOPTS) -L$(NET_LIB_PATH)
 
+INSTALL = install
+INSTALL_PROGRAM = $(INSTALL) -p    -o root -g root  -m  755
+          
+ifneq (,$(findstring debug,$(DEB_BUILD_OPTIONS)))
+CFLAGS += -g
+endif
+
+ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
+INSTALL_PROGRAM +=
+endif
+
+
 SUBDIRS	= man/ $(NET_LIB_PATH)/
 
 ifeq ($(origin CC), undefined)
@@ -161,10 +173,6 @@
 
 config.h: 	config.in Makefile 
 		@echo "Configuring the Linux net-tools (NET-3 Base Utilities)..." ; echo
-		@if [ config.status -nt config.in ]; \
-			then ./configure.sh config.status; \
-		   else ./configure.sh config.in; \
-		 fi
 
 
 version.h:	Makefile
@@ -223,29 +231,29 @@
 installbin:
 	install -m 0755 -d ${BASEDIR}/sbin
 	install -m 0755 -d ${BASEDIR}/bin
-	install -m 0755 arp        ${BASEDIR}/sbin
-	install -m 0755 hostname   ${BASEDIR}/bin
-	install -m 0755 ifconfig   ${BASEDIR}/sbin
-	install -m 0755 nameif     ${BASEDIR}/sbin
-	install -m 0755 netstat    ${BASEDIR}/bin
-	install -m 0755 plipconfig $(BASEDIR)/sbin
-	install -m 0755 rarp       ${BASEDIR}/sbin
-	install -m 0755 route      ${BASEDIR}/sbin
-	install -m 0755 slattach   $(BASEDIR)/sbin
+	$(INSTALL_PROGRAM) arp        ${BASEDIR}/sbin
+#	$(INSTALL_PROGRAM) hostname   ${BASEDIR}/bin
+	$(INSTALL_PROGRAM) ifconfig   ${BASEDIR}/sbin
+	$(INSTALL_PROGRAM) nameif     ${BASEDIR}/sbin
+	$(INSTALL_PROGRAM) netstat    ${BASEDIR}/bin
+	$(INSTALL_PROGRAM) plipconfig $(BASEDIR)/sbin
+	$(INSTALL_PROGRAM) rarp       ${BASEDIR}/sbin
+	$(INSTALL_PROGRAM) route      ${BASEDIR}/sbin
+	$(INSTALL_PROGRAM) slattach   $(BASEDIR)/sbin
 ifeq ($(HAVE_IP_TOOLS),1)
-	install -m 0755 ipmaddr    $(BASEDIR)/sbin
-	install -m 0755 iptunnel   $(BASEDIR)/sbin
+	$(INSTALL_PROGRAM) ipmaddr    $(BASEDIR)/sbin
+	$(INSTALL_PROGRAM) iptunnel   $(BASEDIR)/sbin
 endif
 ifeq ($(HAVE_MII),1)
-	install -m 0755 mii-tool   $(BASEDIR)/sbin
-endif
-	ln -fs hostname $(BASEDIR)/bin/dnsdomainname
-	ln -fs hostname $(BASEDIR)/bin/ypdomainname
-	ln -fs hostname $(BASEDIR)/bin/nisdomainname
-	ln -fs hostname $(BASEDIR)/bin/domainname
-ifeq ($(HAVE_AFDECnet),1)
-	ln -fs hostname $(BASEDIR)/bin/nodename
+	$(INSTALL_PROGRAM) mii-tool   $(BASEDIR)/sbin
 endif
+#	ln -fs hostname $(BASEDIR)/bin/dnsdomainname
+#	ln -fs hostname $(BASEDIR)/bin/ypdomainname
+#	ln -fs hostname $(BASEDIR)/bin/nisdomainname
+#	ln -fs hostname $(BASEDIR)/bin/domainname
+#ifeq ($(HAVE_AFDECnet),1)
+#	ln -fs hostname $(BASEDIR)/bin/nodename
+#endif
 
 savebin:
 	@for i in ${BASEDIR}/sbin/arp ${BASEDIR}/sbin/ifconfig \
diff -Naur net-tools-1.60/man/de_DE/netstat.8 net-tools-1.60-robox-linux/man/de_DE/netstat.8
--- net-tools-1.60/man/de_DE/netstat.8	1999-03-15 21:31:20.000000000 +0100
+++ net-tools-1.60-robox-linux/man/de_DE/netstat.8	2008-04-14 15:57:57.000000000 +0200
@@ -9,7 +9,7 @@
 .\" Modified: Tuan Hoang tuan@optimus.mitre.org 
 .\"
 .\"
-.TH NETSTAT 8 "6. M\(:arz 1999" "net-tools" "Handbuch f\(:ur Linuxprogrammierer"
+.TH NETSTAT 8 "24. November 2001" "net-tools" "Handbuch f\(:ur Linuxprogrammierer"
 
 .SH NAME
 netstat \- Anzeige von Netzwerksverbindungen, Routentabellen, Schnittstellenstatistiken, maskierten Verbindungen, Netlink-Nachrichten und Mitgliedschaft in Multicastgruppen
@@ -44,7 +44,6 @@
 .B netstat
 .RB [ \-veenpac ]
 .RB { \-\-interfaces | \-i }
-.RI [ Schnittstelle ]
 
 .PP
 
@@ -103,11 +102,10 @@
 Wegen Details siehe
 .BR route (8).
 
-.SS "\-i, \-\-interface \fISchnittstelle\fI"
+.SS "\-i, \-\-interface"
 Wird die
 .BR -i ", " --interfaces
-Option verwendet,  so wird eine Tabelle aller (oder der angegebenen
-.IR Schnittstellen ) 
+Option verwendet,  so wird eine Tabelle aller Schnittstellen
 ausgedruckt.  Die Ausgabe ist im Format von
 .B "ifconfig -e"
 und wird in
diff -Naur net-tools-1.60/man/en_US/ifconfig.8 net-tools-1.60-robox-linux/man/en_US/ifconfig.8
--- net-tools-1.60/man/en_US/ifconfig.8	2000-10-28 12:59:42.000000000 +0200
+++ net-tools-1.60-robox-linux/man/en_US/ifconfig.8	2008-04-14 15:57:57.000000000 +0200
@@ -1,4 +1,4 @@
-.TH IFCONFIG 8 "14 August 2000" "net-tools" "Linux Programmer's Manual"
+.TH IFCONFIG 8 "11 Nov 2001" "net-tools" "Linux Programmer's Manual"
 .SH NAME
 ifconfig \- configure a network interface
 .SH SYNOPSIS
@@ -176,8 +176,15 @@
 .BR ipchains(8)
 command.
 .LP
-Interrupt problems with Ethernet device drivers fail with EAGAIN. See
-.I http://cesdis.gsfc.nasa.gov/linux/misc/irq-conflict.html
+Since net-tools 1.60-4 ifconfig is printing byte counters and human readable
+counters with IEC 60027-2 units. So 1 KiB are 2^10 byte. Note, the numbers 
+are truncated to one decimal (which can by quite a large error if you 
+consider 0.1 PiB is 112.589.990.684.262 bytes :)
+.LP
+Interrupt problems with Ethernet device drivers fail with EAGAIN
+.I (SIOCSIIFLAGS: Resource temporarily unavailable)
+it is most likely a interrupt conflict. See
+.I http://www.scyld.com/expert/irq-conflict.html
 for more information.
 .SH FILES
 .I /proc/net/socket 
@@ -190,6 +197,8 @@
 altered by this command.
 .SH SEE ALSO
 route(8), netstat(8), arp(8), rarp(8), ipchains(8)
+.br
+http://physics.nist.gov/cuu/Units/binary.html - Prefixes for binary multiples
 .SH AUTHORS
 Fred N. van Kempen, <waltje@uwalt.nl.mugnet.org>
 .br
@@ -197,4 +206,6 @@
 .br
 Phil Blundell, <Philip.Blundell@pobox.com>
 .br
-Andi Kleen
+Andi Kleen; 
+.br
+Bernd Eckenfels, <net-tools@lina.inka.de>
diff -Naur net-tools-1.60/man/en_US/mii-tool.8 net-tools-1.60-robox-linux/man/en_US/mii-tool.8
--- net-tools-1.60/man/en_US/mii-tool.8	2000-05-21 16:21:38.000000000 +0200
+++ net-tools-1.60-robox-linux/man/en_US/mii-tool.8	2008-04-14 15:57:57.000000000 +0200
@@ -1,7 +1,7 @@
 .\" Copyright (C) 2000 David A. Hinds -- dhinds@pcmcia.sourceforge.org
 .\" mii-tool.8 1.5 2000/04/25 22:58:19
 .\"
-.TH MII-TOOL 8 "2000/04/25 22:58:19" "net-tools"
+.TH MII-TOOL 8 "2004/03/28 23:30:00" "net-tools"
 
 .SH NAME
 mii-tool \- view, manipulate media-independent interface status
@@ -72,7 +72,24 @@
 commas.  Valid media are \fB100baseT4\fR, \fB100baseTx-FD\fR,
 \fB100baseTx-HD\fR, \fB10baseT-FD\fR, and \fB10baseT-HD\fR.
 
+.SH Diagnostics
+.TP
+SIOCGMIIPHY on 'eth?' failed: Invalid argument
+If the interface is not running (up), kernel will refuse to report its link state.
+.TP
+SIOCGMIIPHY on 'eth?' failed: Operation not permitted
+Most kernels restrict access to root.
+.TP
+SIOCGMIIPHY on 'eth?' failed: No such device
+This error is shown, if the kernel does not know about the named device.
+.TP
+SIOCGMIIPHY on 'eth?' failed: Operation not supported 
+The interface in question does not support MII queries. Most likely, it does not have 
+MII tranceivers, at all.
+
 .SH AUTHORS
 David Hinds \- dhinds@pcmcia.sourceforge.org
 .br
 Donald Becker \- becker@scyld.com
+.br
+Bernd Eckenfels \- ecki@debian.org
diff -Naur net-tools-1.60/man/en_US/netstat.8 net-tools-1.60-robox-linux/man/en_US/netstat.8
--- net-tools-1.60/man/en_US/netstat.8	2001-01-07 13:43:57.000000000 +0100
+++ net-tools-1.60-robox-linux/man/en_US/netstat.8	2008-04-14 15:57:57.000000000 +0200
@@ -8,7 +8,7 @@
 .\" Modified: Tuan Hoang tqhoang@bigfoot.com 
 .\"
 .\"
-.TH NETSTAT 8 "19 December 2000" "net-tools" "Linux Programmer's Manual"
+.TH NETSTAT 8 "24 November 2001" "net-tools" "Linux Programmer's Manual"
 
 .SH NAME
 netstat \- Print network connections, routing tables, interface statistics, masquerade connections, and multicast memberships
@@ -23,7 +23,7 @@
 .RB [ \-\-listening | \-l ]
 .RB [ \-\-all | \-a ]
 .RB [ \-\-numeric | \-n ]
-.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
+.RB [ \-\-numeric-hosts "] [" \-\-numeric-ports "] [" \-\-numeric-users ]
 .RB [ \-\-symbolic | \-N ]
 .RB [ \-\-extend | \-e  [ \-\-extend | \-e] ]
 .RB [ \-\-timers | \-o ]
@@ -37,31 +37,30 @@
 .RB [ \-\-extend | \-e  [ \-\-extend | \-e] ]
 .RB [ \-\-verbose | \-v ]
 .RB [ \-\-numeric | \-n ]
-.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
+.RB [ \-\-numeric-hosts "] [" \-\-numeric-ports "] [" \-\-numeric-users ]
 .RB [ \-\-continuous | \-c]
 .P
 .B netstat
 .RB { \-\-interfaces | \-i }
-.RI [ iface ]
 .RB [ \-\-all | \-a ]
 .RB [ \-\-extend | \-e  [ \-\-extend | \-e] ]
 .RB [ \-\-verbose | \-v ]
 .RB [ \-\-program | \-p ]
 .RB [ \-\-numeric | \-n ]
-.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
+.RB [ \-\-numeric-hosts "] [" \-\-numeric-ports "] [" \-\-numeric-users ]
 .RB [ \-\-continuous | \-c]
 .P
 .B netstat
 .RB { \-\-groups | \-g }
 .RB [ \-\-numeric | \-n ]
-.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
+.RB [ \-\-numeric-hosts "] [" \-\-numeric-ports "] [" \-\-numeric-users ]
 .RB [ \-\-continuous | \-c]
 .P
 .B netstat
 .RB { \-\-masquerade | \-M }
 .RB [ \-\-extend | \-e ]
 .RB [ \-\-numeric | \-n ]
-.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
+.RB [ \-\-numeric-hosts "] [" \-\-numeric-ports "] [" \-\-numeric-users ]
 .RB [ \-\-continuous | \-c]
 .P
 .B netstat
@@ -101,9 +100,8 @@
 Display the kernel routing tables.
 .SS "\-\-groups , \-g"
 Display multicast group membership information for IPv4 and IPv6.
-.SS "\-\-interface=\fIiface \fR, \fB\-i"
-Display a table of all network interfaces, or the specified
-.IR iface ) .
+.SS "\-\-interface, \-i"
+Display a table of all network interfaces.
 .SS "\-\-masquerade , \-M"
 Display a list of masqueraded connections.
 .SS "\-\-statistics , \-s"
@@ -165,7 +163,7 @@
 .SS "\-a, \-\-all"
 Show both listening and non-listening sockets.  With the
 .B --interfaces
-option, show interfaces that are not marked 
+option, show interfaces that are not up
 .SS "\-F"
 Print routing information from the FIB.  (This is the default.)
 .SS "\-C"
@@ -221,7 +219,7 @@
 The socket is waiting after close to handle packets still in the network.
 .TP
 .I
-CLOSED
+CLOSE
 The socket is not being used.
 .TP
 .I
@@ -438,7 +436,7 @@
 .P
 .SH AUTHORS
 The netstat user interface was written by Fred Baumgarten
-<dc6iq@insu1.etec.uni-karlsruhe.de> the man page basically
+<dc6iq@insu1.etec.uni-karlsruhe.de>, the man page basically
 by Matt Welsh <mdw@tc.cornell.edu>. It was updated by
 Alan Cox <Alan.Cox@linux.org> but could do with a bit more
 work.  It was updated again by Tuan Hoang
diff -Naur net-tools-1.60/mii-tool.c net-tools-1.60-robox-linux/mii-tool.c
--- net-tools-1.60/mii-tool.c	2000-05-21 16:31:17.000000000 +0200
+++ net-tools-1.60-robox-linux/mii-tool.c	2008-04-14 15:57:57.000000000 +0200
@@ -166,7 +166,7 @@
 	s = strtok(arg, ", ");
 	do {
 	    for (i = 0; i < NMEDIA; i++)
-		if (strcasecmp(media[i].name, s) == 0) break;
+		if (s && strcasecmp(media[i].name, s) == 0) break;
 	    if (i == NMEDIA) goto failed;
 	    mask |= media[i].value;
 	} while ((s = strtok(NULL, ", ")) != NULL);
@@ -302,6 +302,7 @@
 	    printf("\n  link partner:%s", media_list(lkpar, 0));
 	printf("\n");
     }
+    fflush(stdout);
     return 0;
 }
 
@@ -329,7 +330,7 @@
 	printf("resetting the transceiver...\n");
 	mdio_write(skfd, MII_BMCR, MII_BMCR_RESET);
     }
-    if (nway_advertise) {
+    if (nway_advertise > 0) {
 	mdio_write(skfd, MII_ANAR, nway_advertise | 1);
 	opt_restart = 1;
     }
@@ -378,18 +379,17 @@
 
 /*--------------------------------------------------------------------*/
 
-const char *usage =
-"usage: %s [-VvRrwl] [-A media,... | -F media] [interface ...]
-       -V, --version               display version information
-       -v, --verbose               more verbose output
-       -R, --reset                 reset MII to poweron state
-       -r, --restart               restart autonegotiation
-       -w, --watch                 monitor for link status changes
-       -l, --log                   with -w, write events to syslog
-       -A, --advertise=media,...   advertise only specified media
-       -F, --force=media           force specified media technology
-media: 100baseT4, 100baseTx-FD, 100baseTx-HD, 10baseT-FD, 10baseT-HD,
-       (to advertise both HD and FD) 100baseTx, 10baseT\n";
+const char *usage = "usage: %s [-VvRrwl] [-A media,... | -F media] [interface ...]\n"
+"       -V, --version               display version information\n"
+"       -v, --verbose               more verbose output\n"
+"       -R, --reset                 reset MII to poweron state\n"
+"       -r, --restart               restart autonegotiation\n"
+"       -w, --watch                 monitor for link status changes\n"
+"       -l, --log                   with -w, write events to syslog\n"
+"       -A, --advertise=media,...   advertise only specified media\n"
+"       -F, --force=media           force specified media technology\n"
+"media: 100baseT4, 100baseTx-FD, 100baseTx-HD, 10baseT-FD, 10baseT-HD,\n"
+"       (to advertise both HD and FD) 100baseTx, 10baseT\n";
 
 int main(int argc, char **argv)
 {
@@ -411,6 +411,10 @@
 	}
     /* Check for a few inappropriate option combinations */
     if (opt_watch) verbose = 0;
+
+    if ((nway_advertise < 0) || (fixed_speed < 0))
+    	return 2;
+
     if (errflag || (fixed_speed & (fixed_speed-1)) ||
 	(fixed_speed && (opt_restart || nway_advertise))) {
 	fprintf(stderr, usage, argv[0]);
@@ -425,6 +429,9 @@
 	perror("socket");
 	exit(-1);
     }
+    
+    if (verbose > 1)
+    	printf("Using SIOCGMIIPHY=0x%x\n", SIOCGMIIPHY);	
 
     /* No remaining args means show all interfaces. */
     if (optind == argc) {
diff -Naur net-tools-1.60/nameif.c net-tools-1.60-robox-linux/nameif.c
--- net-tools-1.60/nameif.c	2000-10-18 19:26:29.000000000 +0200
+++ net-tools-1.60-robox-linux/nameif.c	2008-04-14 15:57:57.000000000 +0200
@@ -3,7 +3,7 @@
  * Writen 2000 by Andi Kleen.
  * Subject to the Gnu Public License, version 2.  
  * TODO: make it support token ring etc.
- * $Id: nameif.c,v 1.1 2000/10/18 17:26:29 ak Exp $
+ * $Id: nameif.c,v 1.3 2003/03/06 23:26:52 ecki Exp $
  */ 
 #ifndef _GNU_SOURCE 
 #define _GNU_SOURCE
@@ -117,7 +117,8 @@
 }
 
 struct change { 
-	struct change *next,**pprev;
+	struct change *next;
+	int found;
 	char ifname[IFNAMSIZ+1];
 	unsigned char mac[6];
 }; 
@@ -139,10 +140,7 @@
 			ch->ifname, pos); 
 	if (parsemac(p,ch->mac) < 0) 
 		complain(_("cannot parse MAC `%s' at %s"), p, pos); 
-	if (clist) 
-		clist->pprev = &ch->next;
 	ch->next = clist;
-	ch->pprev = &clist;
 	clist = ch;
 	return 0; 
 }
@@ -200,7 +198,7 @@
 
 void usage(void)
 {
-	fprintf(stderr, _("usage: nameif [-c configurationfile] [-s] {ifname macaddress}")); 
+	fprintf(stderr, _("usage: nameif [-c configurationfile] [-s] {ifname macaddress}\n")); 
 	exit(1); 
 }
 
@@ -277,21 +275,21 @@
 		ch = lookupmac(mac); 
 		if (!ch) 
 			continue;
-			
-		*ch->pprev = ch->next;
+		
+		ch->found = 1;	
 		if (strcmp(p, ch->ifname)) { 
 			if (setname(p, ch->ifname) < 0)  
 				complain(_("cannot change name of %s to %s: %s"),
 						p, ch->ifname, strerror(errno)); 
 		} 
-		free(ch);
 	} 
 	fclose(ifh); 
 	
 	while (clist) { 
 		struct change *ch = clist;
 		clist = clist->next;
-		warning(_("interface '%s' not found"), ch->ifname); 
+		if (!ch->found)
+			warning(_("interface '%s' not found"), ch->ifname); 
 		free(ch); 
 	}
 
diff -Naur net-tools-1.60/netstat.c net-tools-1.60-robox-linux/netstat.c
--- net-tools-1.60/netstat.c	2001-04-15 16:41:17.000000000 +0200
+++ net-tools-1.60-robox-linux/netstat.c	2008-04-14 15:57:57.000000000 +0200
@@ -6,7 +6,7 @@
  *              NET-3 Networking Distribution for the LINUX operating
  *              system.
  *
- * Version:     $Id: netstat.c,v 1.43 2001/04/15 14:41:17 pb Exp $
+ * Version:     $Id: netstat.c,v 1.50 2002/12/10 00:56:41 ecki Exp $
  *
  * Authors:     Fred Baumgarten, <dc6iq@insu1.etec.uni-karlsruhe.de>
  *              Fred N. van Kempen, <waltje@uwalt.nl.mugnet.org>
@@ -216,7 +216,7 @@
 
 static struct prg_node {
     struct prg_node *next;
-    int inode;
+    unsigned long inode;
     char name[PROGNAME_WIDTH];
 } *prg_hash[PRG_HASH_SIZE];
 
@@ -249,7 +249,7 @@
 /* NOT working as of glibc-2.0.7: */
 #undef  DIRENT_HAVE_D_TYPE_WORKS
 
-static void prg_cache_add(int inode, char *name)
+static void prg_cache_add(unsigned long inode, char *name)
 {
     unsigned hi = PRG_HASHIT(inode);
     struct prg_node **pnp,*pn;
@@ -272,7 +272,7 @@
     strcpy(pn->name,name);
 }
 
-static const char *prg_cache_get(int inode)
+static const char *prg_cache_get(unsigned long inode)
 {
     unsigned hi=PRG_HASHIT(inode);
     struct prg_node *pn;
@@ -295,16 +295,18 @@
     prg_cache_loaded=0;
 }
 
-static void extract_type_1_socket_inode(const char lname[], long * inode_p) {
+static int extract_type_1_socket_inode(const char lname[], unsigned long * inode_p) {
 
     /* If lname is of the form "socket:[12345]", extract the "12345"
        as *inode_p.  Otherwise, return -1 as *inode_p.
        */
 
-    if (strlen(lname) < PRG_SOCKET_PFXl+3) *inode_p = -1;
-    else if (memcmp(lname, PRG_SOCKET_PFX, PRG_SOCKET_PFXl)) *inode_p = -1;
-    else if (lname[strlen(lname)-1] != ']') *inode_p = -1;
-    else {
+    if (strlen(lname) < PRG_SOCKET_PFXl+3) return(-1);
+    
+    if (memcmp(lname, PRG_SOCKET_PFX, PRG_SOCKET_PFXl)) return(-1);
+    if (lname[strlen(lname)-1] != ']') return(-1);
+
+    {
         char inode_str[strlen(lname + 1)];  /* e.g. "12345" */
         const int inode_str_len = strlen(lname) - PRG_SOCKET_PFXl - 1;
         char *serr;
@@ -313,37 +315,41 @@
         inode_str[inode_str_len] = '\0';
         *inode_p = strtol(inode_str,&serr,0);
         if (!serr || *serr || *inode_p < 0 || *inode_p >= INT_MAX) 
-            *inode_p = -1;
+            return(-1);
     }
+    return(0);
 }
 
 
 
-static void extract_type_2_socket_inode(const char lname[], long * inode_p) {
+static int extract_type_2_socket_inode(const char lname[], unsigned long * inode_p) {
 
     /* If lname is of the form "[0000]:12345", extract the "12345"
        as *inode_p.  Otherwise, return -1 as *inode_p.
        */
 
-    if (strlen(lname) < PRG_SOCKET_PFX2l+1) *inode_p = -1;
-    else if (memcmp(lname, PRG_SOCKET_PFX2, PRG_SOCKET_PFX2l)) *inode_p = -1;
-    else {
+    if (strlen(lname) < PRG_SOCKET_PFX2l+1) return(-1);
+    if (memcmp(lname, PRG_SOCKET_PFX2, PRG_SOCKET_PFX2l)) return(-1);
+
+    {
         char *serr;
 
         *inode_p=strtol(lname + PRG_SOCKET_PFX2l,&serr,0);
         if (!serr || *serr || *inode_p < 0 || *inode_p >= INT_MAX) 
-            *inode_p = -1;
+            return(-1);
     }
+    return(0);
 }
 
 
 
+
 static void prg_cache_load(void)
 {
     char line[LINE_MAX],eacces=0;
     int procfdlen,fd,cmdllen,lnamelen;
     char lname[30],cmdlbuf[512],finbuf[PROGNAME_WIDTH];
-    long inode;
+    unsigned long inode;
     const char *cs,*cmdlp;
     DIR *dirproc=NULL,*dirfd=NULL;
     struct dirent *direproc,*direfd;
@@ -386,11 +392,9 @@
 	    lnamelen=readlink(line,lname,sizeof(lname)-1);
             lname[lnamelen] = '\0';  /*make it a null-terminated string*/
 
-            extract_type_1_socket_inode(lname, &inode);
-
-            if (inode < 0) extract_type_2_socket_inode(lname, &inode);
-
-            if (inode < 0) continue;
+            if (extract_type_1_socket_inode(lname, &inode) < 0)
+              if (extract_type_2_socket_inode(lname, &inode) < 0)
+                continue;
 
 	    if (!cmdlp) {
 		if (procfdlen - PATH_FD_SUFFl + PATH_CMDLINEl >= 
@@ -530,7 +534,7 @@
 	    printf("%-10s ", pw->pw_name);
 	else
 	    printf("%-10d ", uid);
-	printf("%-10ld ",inode);
+	printf("%-10lu ",inode);
     }
     if (flag_prg)
 	printf("%-" PROGNAME_WIDTHs "s",prg_cache_get(inode));
@@ -705,6 +709,7 @@
     unsigned long rxq, txq, time_len, retr, inode;
     int num, local_port, rem_port, d, state, uid, timer_run, timeout;
     char rem_addr[128], local_addr[128], timers[64], buffer[1024], more[512];
+    char *protname;
     struct aftype *ap;
 #if HAVE_AFINET6
     struct sockaddr_in6 localaddr, remaddr;
@@ -719,12 +724,13 @@
 	return;
 
     num = sscanf(line,
-    "%d: %64[0-9A-Fa-f]:%X %64[0-9A-Fa-f]:%X %X %lX:%lX %X:%lX %lX %d %d %ld %512s\n",
+    "%d: %64[0-9A-Fa-f]:%X %64[0-9A-Fa-f]:%X %X %lX:%lX %X:%lX %lX %d %d %lu %512s\n",
 		 &d, local_addr, &local_port, rem_addr, &rem_port, &state,
 		 &txq, &rxq, &timer_run, &time_len, &retr, &uid, &timeout, &inode, more);
 
     if (strlen(local_addr) > 8) {
 #if HAVE_AFINET6
+	protname = "tcp6";
 	/* Demangle what the kernel gives us */
 	sscanf(local_addr, "%08X%08X%08X%08X",
 	       &in6.s6_addr32[0], &in6.s6_addr32[1],
@@ -740,6 +746,7 @@
 	remaddr.sin6_family = AF_INET6;
 #endif
     } else {
+	protname = "tcp";
 	sscanf(local_addr, "%X",
 	       &((struct sockaddr_in *) &localaddr)->sin_addr.s_addr);
 	sscanf(rem_addr, "%X",
@@ -813,8 +820,8 @@
 			 timer_run, (double) time_len / HZ, retr, timeout);
 		break;
 	    }
-	printf("tcp   %6ld %6ld %-23s %-23s %-12s",
-	       rxq, txq, local_addr, rem_addr, _(tcp_state[state]));
+	printf("%-4s  %6ld %6ld %-*s %-*s %-11s",
+	       protname, rxq, txq, netmax(23,strlen(local_addr)), local_addr, netmax(23,strlen(rem_addr)), rem_addr, _(tcp_state[state]));
 
 	finish_this_one(uid,inode,timers);
     }
@@ -831,6 +838,7 @@
     char buffer[8192], local_addr[64], rem_addr[64];
     char *udp_state, timers[64], more[512];
     int num, local_port, rem_port, d, state, timer_run, uid, timeout;
+    char *protname;
 #if HAVE_AFINET6
     struct sockaddr_in6 localaddr, remaddr;
     char addr6[INET6_ADDRSTRLEN];
@@ -847,13 +855,14 @@
 
     more[0] = '\0';
     num = sscanf(line,
-		 "%d: %64[0-9A-Fa-f]:%X %64[0-9A-Fa-f]:%X %X %lX:%lX %X:%lX %lX %d %d %ld %512s\n",
+		 "%d: %64[0-9A-Fa-f]:%X %64[0-9A-Fa-f]:%X %X %lX:%lX %X:%lX %lX %d %d %lu %512s\n",
 		 &d, local_addr, &local_port,
 		 rem_addr, &rem_port, &state,
 	  &txq, &rxq, &timer_run, &time_len, &retr, &uid, &timeout, &inode, more);
 
     if (strlen(local_addr) > 8) {
 #if HAVE_AFINET6
+	protname="udp6";
 	sscanf(local_addr, "%08X%08X%08X%08X",
 	       &in6.s6_addr32[0], &in6.s6_addr32[1],
 	       &in6.s6_addr32[2], &in6.s6_addr32[3]);
@@ -868,6 +877,7 @@
 	remaddr.sin6_family = AF_INET6;
 #endif
     } else {
+        protname="udp";
 	sscanf(local_addr, "%X",
 	       &((struct sockaddr_in *) &localaddr)->sin_addr.s_addr);
 	sscanf(rem_addr, "%X",
@@ -953,8 +963,8 @@
 			 retr, timeout);
 		break;
 	    }
-	printf("udp   %6ld %6ld %-23s %-23s %-12s",
-	       rxq, txq, local_addr, rem_addr, udp_state);
+	printf("%-4s  %6ld %6ld %-23s %-23s %-11s",
+	       protname, rxq, txq, local_addr, rem_addr, udp_state);
 
 	finish_this_one(uid,inode,timers);
     }
@@ -971,6 +981,7 @@
     char buffer[8192], local_addr[64], rem_addr[64];
     char timers[64], more[512];
     int num, local_port, rem_port, d, state, timer_run, uid, timeout;
+    char *protname;
 #if HAVE_AFINET6
     struct sockaddr_in6 localaddr, remaddr;
     char addr6[INET6_ADDRSTRLEN];
@@ -987,12 +998,13 @@
 
     more[0] = '\0';
     num = sscanf(line,
-		 "%d: %64[0-9A-Fa-f]:%X %64[0-9A-Fa-f]:%X %X %lX:%lX %X:%lX %lX %d %d %ld %512s\n",
+		 "%d: %64[0-9A-Fa-f]:%X %64[0-9A-Fa-f]:%X %X %lX:%lX %X:%lX %lX %d %d %lu %512s\n",
 		 &d, local_addr, &local_port, rem_addr, &rem_port, &state,
 	  &txq, &rxq, &timer_run, &time_len, &retr, &uid, &timeout, &inode, more);
 
     if (strlen(local_addr) > 8) {
 #if HAVE_AFINET6
+	protname = "raw6";
 	sscanf(local_addr, "%08X%08X%08X%08X",
 	       &in6.s6_addr32[0], &in6.s6_addr32[1],
            &in6.s6_addr32[2], &in6.s6_addr32[3]);
@@ -1007,6 +1019,7 @@
 	remaddr.sin6_family = AF_INET6;
 #endif
     } else {
+        protname = "raw";
 	sscanf(local_addr, "%X",
 	       &((struct sockaddr_in *) &localaddr)->sin_addr.s_addr);
 	sscanf(rem_addr, "%X",
@@ -1074,8 +1087,8 @@
 			 retr, timeout);
 		break;
 	    }
-	printf("raw   %6ld %6ld %-23s %-23s %-12d",
-	       rxq, txq, local_addr, rem_addr, state);
+	printf("%-4s  %6ld %6ld %-23s %-23s %-11d",
+	       protname, rxq, txq, local_addr, rem_addr, state);
 
 	finish_this_one(uid,inode,timers);
     }
@@ -1099,9 +1112,9 @@
     static int has = 0;
     char path[MAXPATHLEN], ss_flags[32];
     char *ss_proto, *ss_state, *ss_type;
-    int num, state, type, inode;
+    int num, state, type;
     void *d;
-    unsigned long refcnt, proto, flags;
+    unsigned long refcnt, proto, flags, inode;
 
     if (nr == 0) {
 	if (strstr(line, "Inode"))
@@ -1109,14 +1122,14 @@
 	return;
     }
     path[0] = '\0';
-    num = sscanf(line, "%p: %lX %lX %lX %X %X %d %s",
+    num = sscanf(line, "%p: %lX %lX %lX %X %X %lu %s",
 		 &d, &refcnt, &proto, &flags, &type, &state, &inode, path);
     if (num < 6) {
 	fprintf(stderr, _("warning, got bogus unix line.\n"));
 	return;
     }
     if (!(has & HAS_INODE))
-	snprintf(path,sizeof(path),"%d",inode);
+	snprintf(path,sizeof(path),"%lu",inode);
 
     if (!flag_all) {
     	if ((state == SS_UNCONNECTED) && (flags & SO_ACCEPTCON)) {
@@ -1208,9 +1221,9 @@
     printf("%-5s %-6ld %-11s %-10s %-13s ",
 	   ss_proto, refcnt, ss_flags, ss_type, ss_state);
     if (has & HAS_INODE)
-	printf("%-6d ",inode);
+	printf("%-8lu ",inode);
     else
-	printf("-      ");
+	printf("-        ");
     if (flag_prg)
 	printf("%-" PROGNAME_WIDTHs "s",(has & HAS_INODE?prg_cache_get(inode):"-"));
     puts(path);
@@ -1457,7 +1470,7 @@
 	exit(1);
     }
     if (flag_cnt)
-	free_interface_list();
+	if_cache_free();
     else {
 	close(skfd);
 	skfd = -1;
@@ -1503,7 +1516,7 @@
     fprintf(stderr, _("        -C, --cache              display routing cache instead of FIB\n\n"));
 
     fprintf(stderr, _("  <Socket>={-t|--tcp} {-u|--udp} {-w|--raw} {-x|--unix} --ax25 --ipx --netrom\n"));
-    fprintf(stderr, _("  <AF>=Use '-A <af>' or '--<af>'; default: %s\n"), DFLT_AF);
+    fprintf(stderr, _("  <AF>=Use '-6|-4' or '-A <af>' or '--<af>'; default: %s\n"), DFLT_AF);
     fprintf(stderr, _("  List of possible address families (which support routing):\n"));
     print_aflist(1); /* 1 = routeable */
     exit(E_USAGE);
@@ -1514,7 +1527,7 @@
  (int argc, char *argv[]) {
     int i;
     int lop;
-    struct option longopts[] =
+    static struct option longopts[] =
     {
 	AFTRANS_OPTS,
 	{"version", 0, 0, 'V'},
@@ -1556,7 +1569,7 @@
     getroute_init();		/* Set up AF routing support */
 
     afname[0] = '\0';
-    while ((i = getopt_long(argc, argv, "MCFA:acdegphinNorstuVv?wxl", longopts, &lop)) != EOF)
+    while ((i = getopt_long(argc, argv, "MCFA:acdegphinNorstuVv?wxl64", longopts, &lop)) != EOF)
 	switch (i) {
 	case -1:
 	    break;
@@ -1624,6 +1637,14 @@
 	case 'o':
 	    flag_opt++;
 	    break;
+	case '6':
+	    if (aftrans_opt("inet6"))
+		exit(1);
+	    break;
+	case '4':
+	    if (aftrans_opt("inet"))
+		exit(1);
+	    break;
 	case 'V':
 	    version();
 	    /*NOTREACHED */
diff -Naur net-tools-1.60/po/de.po net-tools-1.60-robox-linux/po/de.po
--- net-tools-1.60/po/de.po	2000-10-28 12:59:43.000000000 +0200
+++ net-tools-1.60-robox-linux/po/de.po	2008-04-14 15:57:57.000000000 +0200
@@ -1,4 +1,4 @@
-# $Id: de.po,v 1.10 2000/08/01 03:19:48 ecki Exp $
+# $Id: de.po,v 1.11 2003/10/25 21:15:09 ecki Exp $
 # German translation for net-tools 1.51
 # Copyright (C) 1999 Ralf Bächle <ralf@gnu.org>
 msgid ""
@@ -64,7 +64,7 @@
 
 #: ../arp.c:467
 msgid "(incomplete)"
-msgstr "(unvollsändig)"
+msgstr "(unvollständig)"
 
 #: ../arp.c:484
 #, c-format
diff -Naur net-tools-1.60/README net-tools-1.60-robox-linux/README
--- net-tools-1.60/README	2001-04-15 16:34:43.000000000 +0200
+++ net-tools-1.60-robox-linux/README	2008-04-14 15:57:57.000000000 +0200
@@ -10,7 +10,7 @@
 subsystem of the Linux kernel.  This includes arp, hostname, ifconfig,
 netstat, rarp and route.  Additionally, this package contains
 utilities relating to particular network hardware types (plipconfig,
-slattach) and advanced aspects of IP configuration (iptunnel,
+slattach, mii-tool) and advanced aspects of IP configuration (iptunnel,
 ipmaddr).
 
 Please include the output of "program --version" when reporting bugs.
@@ -24,8 +24,6 @@
 
    INSTALLING		Installation instructions.
 
-   COPYING		Your free copy of the GNU Public License.
-
    TODO			Some things that need to be done.
 
 
diff -Naur net-tools-1.60/route.c net-tools-1.60-robox-linux/route.c
--- net-tools-1.60/route.c	2001-04-15 16:41:17.000000000 +0200
+++ net-tools-1.60-robox-linux/route.c	2008-04-14 15:57:57.000000000 +0200
@@ -2,7 +2,7 @@
  * route        This file contains an implementation of the command
  *              that manages the IP routing table in the kernel.
  *
- * Version:     $Id: route.c,v 1.9 2001/04/15 14:41:17 pb Exp $
+ * Version:     $Id: route.c,v 1.10 2002/07/30 05:24:20 ecki Exp $
  *
  * Maintainer:  Bernd 'eckes' Eckenfels, <net-tools@lina.inka.de>
  *
@@ -142,7 +142,7 @@
     }
 
     /* Fetch the command-line arguments. */
-    while ((i = getopt_long(argc, argv, "A:eCFhnNVv?", longopts, &lop)) != EOF)
+    while ((i = getopt_long(argc, argv, "A:eCFhnN64Vv?", longopts, &lop)) != EOF)
 	switch (i) {
 	case -1:
 	    break;
@@ -176,6 +176,14 @@
 	    if ((i = aftrans_opt(optarg)))
 		exit(i);
 	    break;
+	case '6':
+	    if ((i = aftrans_opt("inet6")))
+		exit(i);
+	    break;
+	case '4':
+	    if ((i = aftrans_opt("inet")))
+		exit(i);
+	    break;
 	case 'V':
 	    version();
 	case 'h':
diff -Naur net-tools-1.60/statistics.c net-tools-1.60-robox-linux/statistics.c
--- net-tools-1.60/statistics.c	2001-02-02 19:01:23.000000000 +0100
+++ net-tools-1.60-robox-linux/statistics.c	2008-04-14 15:57:57.000000000 +0200
@@ -1,6 +1,6 @@
 /*
  * Copyright 1997,1999,2000 Andi Kleen. Subject to the GPL. 
- * $Id: statistics.c,v 1.14 2001/02/02 18:01:23 pb Exp $
+ * $Id: statistics.c,v 1.17 2002/04/28 15:41:01 ak Exp $
  * 19980630 - i18n - Arnaldo Carvalho de Melo <acme@conectiva.com.br> 
  * 19981113 - i18n fixes - Arnaldo Carvalho de Melo <acme@conectiva.com.br> 
  * 19990101 - added net/netstat, -t, -u, -w supprt - Bernd Eckenfels 
@@ -175,16 +175,54 @@
     { "ListenDrops", N_("%u SYNs to LISTEN sockets ignored"), opt_number },
     { "TCPPrequeued", N_("%u packets directly queued to recvmsg prequeue."), 
       opt_number },
-    { "TCPDirectCopyFromBacklog", N_("%u packets directly received"
+    { "TCPDirectCopyFromBacklog", N_("%u of bytes directly received"
 				     " from backlog"), opt_number },
-    { "TCPDirectCopyFromPrequeue", N_("%u packets directly received"
+    { "TCPDirectCopyFromPrequeue", N_("%u of bytes directly received"
 				      " from prequeue"), opt_number },
     { "TCPPrequeueDropped", N_("%u packets dropped from prequeue"), opt_number },
-    { "TCPHPHits", N_("%u packets header predicted"), number },
+    { "TCPHPHits", N_("%u packet headers predicted"), number },
     { "TCPHPHitsToUser", N_("%u packets header predicted and "
 			    "directly queued to user"), opt_number },
     { "SockMallocOOM", N_("Ran %u times out of system memory during " 
 			  "packet sending"), opt_number }, 
+    { "TCPPureAcks", N_("%u acknowledgments not containing data received"), opt_number },
+    { "TCPHPAcks", N_("%u predicted acknowledgments"), opt_number },
+    { "TCPRenoRecovery", N_("%u times recovered from packet loss due to fast retransmit"), opt_number },
+    { "TCPSackRecovery", N_("%u times recovered from packet loss due to SACK data"), opt_number },
+    { "TCPSACKReneging", N_("%u bad SACKs received"), opt_number },
+    { "TCPFACKReorder", N_("Detected reordering %u times using FACK"), opt_number },
+    { "TCPSACKReorder", N_("Detected reordering %u times using SACK"), opt_number },
+    { "TCPTSReorder", N_("Detected reordering %u times using time stamp"), opt_number },
+    { "TCPRenoReorder", N_("Detected reordering %u times using reno fast retransmit"), opt_number },
+    { "TCPFullUndo", N_("%u congestion windows fully recovered"), opt_number }, 
+    { "TCPPartialUndo", N_("%u congestion windows partially recovered using Hoe heuristic"), opt_number },
+    { "TCPDSackUndo", N_("%u congestion window recovered using DSACK"), opt_number },
+    { "TCPLossUndo", N_("%u congestion windows recovered after partial ack"), opt_number },
+    { "TCPLostRetransmits", N_("%u retransmits lost"), opt_number },
+    { "TCPRenoFailures",  N_("%u timeouts after reno fast retransmit"), opt_number },
+    { "TCPSackFailures",  N_("%u timeouts after SACK recovery"), opt_number },
+    { "TCPLossFailures",  N_("%u timeouts in loss state"), opt_number },
+    { "TCPFastRetrans", N_("%u fast retransmits"), opt_number },
+    { "TCPForwardRetrans", N_("%u forward retransmits"), opt_number }, 
+    { "TCPSlowStartRetrans", N_("%u retransmits in slow start"), opt_number },
+    { "TCPTimeouts", N_("%u other TCP timeouts"), opt_number },
+    { "TCPRenoRecoveryFailed", N_("%u reno fast retransmits failed"), opt_number },
+    { "TCPSackRecoveryFail", N_("%u sack retransmits failed"), opt_number },
+    { "TCPSchedulerFailed", N_("%u times receiver scheduled too late for direct processing"), opt_number },
+    { "TCPRcvCollapsed", N_("%u packets collapsed in receive queue due to low socket buffer"), opt_number },
+    { "TCPDSACKOldSent", N_("%u DSACKs sent for old packets"), opt_number },
+    { "TCPDSACKOfoSent", N_("%u DSACKs sent for out of order packets"), opt_number },
+    { "TCPDSACKRecv", N_("%u DSACKs received"), opt_number },
+    { "TCPDSACKOfoRecv", N_("%u DSACKs for out of order packets received"), opt_number },
+    { "TCPAbortOnSyn", N_("%u connections reset due to unexpected SYN"), opt_number },
+    { "TCPAbortOnData", N_("%u connections reset due to unexpected data"), opt_number },
+    { "TCPAbortOnClose", N_("%u connections reset due to early user close"), opt_number },
+    { "TCPAbortOnMemory", N_("%u connections aborted due to memory pressure"), opt_number },
+    { "TCPAbortOnTimeout", N_("%u connections aborted due to timeout"), opt_number },
+    { "TCPAbortOnLinger", N_("%u connections aborted after user close in linger timeout"), opt_number },
+    { "TCPAbortFailed", N_("%u times unabled to send RST due to no memory"), opt_number }, 
+    { "TCPMemoryPressures", N_("TCP ran low on memory %u times"), opt_number }, 
+    { "TCPLoss", N_("%u TCP data loss events") },
 };
 
 struct tabtab {
@@ -222,7 +260,8 @@
 	    ent = bsearch(&key, tab->tab, tab->size / sizeof(struct entry),
 			  sizeof(struct entry), cmpentries);
     if (!ent) {			/* try our best */
-	printf("%*s%s: %d\n", states[state].indent, "", title, val);
+	if (val) 
+		printf("%*s%s: %d\n", states[state].indent, "", title, val);
 	return;
     }
     type = ent->type;
