diff -Naur carmen-0.7-beta/src/configure carmen-0.7-beta-config/src/configure
--- carmen-0.7-beta/src/configure	2007-09-19 10:13:24.000000000 +0200
+++ carmen-0.7-beta-config/src/configure	2008-04-17 16:44:30.000000000 +0200
@@ -144,11 +144,12 @@
 $python = ask_python() unless ($nopython or !$swig);
 $nopython = !$python;
 
-$oldlaser = ask_oldlaser() unless ($arg_oldlaser);
+#$oldlaser = ask_oldlaser() unless ($arg_oldlaser);
 set_laser_link();
 
-my $install_path = get_install_path() unless ($install_path);
-my @robots = get_robot_types($os_type, $kernel, $canlib);
+$install_path = get_install_path() unless ($install_path);
+#my @robots = get_robot_types($os_type, $kernel, $canlib);
+my @robots;
 
 open OUTPUT, "> Makefile.vars";
 
