diff -Naur carmen-0.7-beta/src/laser-new/sick_laser.c carmen-0.7-beta-sick-baudrate/src/laser-new/sick_laser.c
--- carmen-0.7-beta/src/laser-new/sick_laser.c	2006-11-14 14:59:52.000000000 +0100
+++ carmen-0.7-beta-sick-baudrate/src/laser-new/sick_laser.c	2008-04-28 18:20:44.000000000 +0200
@@ -348,10 +348,10 @@
 
 
 
-int sick_test_baudrate(sick_laser_t* sick, int max_retries){
+int sick_test_baudrate(sick_laser_t* sick, int max_retries, int baudrate){
   //	printf ("Test baudrate ... ");
-  int baudrates[]={9600, 19200, 38400, 500000};
-  int baudrates_no=4;
+  int baudrates[]={baudrate, 9600, 19200, 38400, 500000};
+  int baudrates_no=5;
   unsigned char reply[8192];
   int i;
   sick->baudrate=0;
@@ -690,7 +690,7 @@
   serial_ClearInputBuffer(sick->fd);
   
   fprintf(stderr,"  Querying baudrate ................ ");
-  currentbrate=sick_test_baudrate(sick, 8);
+  currentbrate=sick_test_baudrate(sick, 8, baudrate);
   if (currentbrate<0){
     fprintf(stderr,"error\n  sick_test_baudrate failed!\n");
     return -2;
