diff -ru gcc-4.1.1/configure gcc-4.1.1-xcomp/configure
--- gcc-4.1.1/configure	2005-12-16 13:57:40.000000000 +0100
+++ gcc-4.1.1-xcomp/configure	2007-06-01 20:57:24.000000000 +0200
@@ -939,12 +939,7 @@
 appdirs=""
 
 # Define is_cross_compiler to save on calls to 'test'.
-is_cross_compiler=
-if test x"${host}" = x"${target}" ; then
-  is_cross_compiler=no
-else
-  is_cross_compiler=yes
-fi	
+is_cross_compiler=yes
 
 # Find the build and target subdir names.
  case ${build_alias} in
