diff -Naur iputils-20020927/Makefile iputils-20020927-xcomp/Makefile
--- iputils-20020927/Makefile	2008-04-15 09:11:28.000000000 +0200
+++ iputils-20020927-xcomp/Makefile	2008-04-15 09:23:05.000000000 +0200
@@ -21,11 +21,11 @@
 #options if you compile with libc5, and without a bind>=4.9.4 libresolv
 # NOT AVAILABLE. Please, use libresolv.
 
-CC=gcc
+#CC=gcc
 # What a pity, all new gccs are buggy and -Werror does not work. Sigh.
 #CCOPT=-D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall -g -Werror
 CCOPT=-D_GNU_SOURCE -O2 -Wstrict-prototypes -Wall -g
-CFLAGS=$(CCOPT) $(GLIBCFIX) -I$(KERNEL_INCLUDE) -I../include $(DEFINES) 
+CFLAGS=$(CCOPT) $(GLIBCFIX) -I../include $(DEFINES) 
 
 IPV4_TARGETS=tracepath ping arping
 IPV6_TARGETS=tracepath6 traceroute6 ping6
