diff -Naur linux-2.6.13/arch/ppc/boot/simple/misc.c linux-2.6.13-pcore/arch/ppc/boot/simple/misc.c
--- linux-2.6.13/arch/ppc/boot/simple/misc.c	2008-04-23 11:39:05.000000000 +0200
+++ linux-2.6.13-pcore/arch/ppc/boot/simple/misc.c	2008-04-22 09:50:15.000000000 +0200
@@ -269,11 +269,6 @@
 	return rec;
 }
 
-void __attribute__ ((weak))
-board_isa_init(void)
-{
-}
-
 /* Allow decompress_kernel to be hooked into.  This is the default. */
 void * __attribute__ ((weak))
 load_kernel(unsigned long load_addr, int num_words, unsigned long cksum,
diff -Naur linux-2.6.13/arch/ppc/platforms/pcore.c linux-2.6.13-pcore/arch/ppc/platforms/pcore.c
--- linux-2.6.13/arch/ppc/platforms/pcore.c	2008-04-23 11:39:05.000000000 +0200
+++ linux-2.6.13-pcore/arch/ppc/platforms/pcore.c	2008-04-26 19:26:00.000000000 +0200
@@ -51,13 +51,13 @@
 	 *      A       B       C       D
 	 */
 	{
-		{9,	10,	11,	12},	/* IDSEL 24 - DEC 21554 */
-		{10,	0,	0,	0},	/* IDSEL 25 - DEC 21143 */
-		{11,	12,	9,	10},	/* IDSEL 26 - PMC I */
-		{12,	9,	10,	11},	/* IDSEL 27 - PMC II */
-		{0,	0,	0,	0},	/* IDSEL 28 - unused */
-		{0,	0,	9,	0},	/* IDSEL 29 - unused */
-		{0,	0,	0,	0},	/* IDSEL 30 - Winbond */
+                {9,     10,     11,     12},    /* IDSEL 24 - DEC 21554 PCI-to-PCI bridge */
+                {9,     0,      0,      0},     /* IDSEL 25 - DEC 21143 Ethernet controller */
+                {11,    12,     9,      10},    /* IDSEL 26 - PMC I */
+                {12,    9,      10,     11},    /* IDSEL 27 - PMC II */
+                {0,     0,      0,      0},     /* IDSEL 28 - unused */
+                {0,     0,      0,      0},     /* IDSEL 29 - unused */
+                {0,     0,      0,      0},     /* IDSEL 30 - Winbond */
 		};
 	const long min_idsel = 24, max_idsel = 30, irqs_per_slot = 4;
 	return PCI_IRQ_TABLE_LOOKUP;
@@ -235,8 +235,6 @@
 	else
 		printk("680\n");
 	printk(KERN_INFO "Port by MontaVista Software, Inc. (source@mvista.com)\n");
-	_set_L2CR(L2CR_L2E | _get_L2CR());
-
 }
 
 static void
@@ -244,7 +242,18 @@
 {
 	local_irq_disable();
 	/* Hard reset */
-	writeb(0x11, 0xfe000332);
+        // writeb(0x11, 0xfe000332);
+
+        /* Soft reset */
+        __asm__ __volatile__
+        ("\n\
+         lis     3,0xfff0        \n\
+         ori     3,3,0x0100      \n\
+         mtspr   26,3            \n\
+         li      3,0             \n\
+         mtspr   27,3            \n\
+         rfi                     \n\
+        ");
 	while(1);
 }
 
