diff -Naur linux-2.6.13/arch/ppc/platforms/pcore.c linux-2.6.13-userled/arch/ppc/platforms/pcore.c
--- linux-2.6.13/arch/ppc/platforms/pcore.c	2008-04-23 11:39:05.000000000 +0200
+++ linux-2.6.13-userled/arch/ppc/platforms/pcore.c	2008-04-23 13:27:57.000000000 +0200
@@ -237,6 +237,8 @@
 	printk(KERN_INFO "Port by MontaVista Software, Inc. (source@mvista.com)\n");
 	_set_L2CR(L2CR_L2E | _get_L2CR());
 
+	/* Turn user LED red */
+	writeb(0x02, 0xfe000300);
 }
 
 static void
