diff -Naur linux-2.6.13/arch/ppc/platforms/pcore.c linux-2.6.13-decirq/arch/ppc/platforms/pcore.c
--- linux-2.6.13/arch/ppc/platforms/pcore.c	2005-08-29 01:41:01.000000000 +0200
+++ linux-2.6.13-decirq/arch/ppc/platforms/pcore.c	2008-04-14 19:46:50.000000000 +0200
@@ -51,8 +51,8 @@
 	 *      A       B       C       D
 	 */
 	{
-		{9,	10,	11,	12},	/* IDSEL 24 - DEC 21554 */
-		{10,	0,	0,	0},	/* IDSEL 25 - DEC 21143 */
+		{0,	0,	0,	0},	/* IDSEL 24 - DEC 21554 */
+		{9,	0,	0,	0},	/* IDSEL 25 - DEC 21143 */
 		{11,	12,	9,	10},	/* IDSEL 26 - PMC I */
 		{12,	9,	10,	11},	/* IDSEL 27 - PMC II */
 		{0,	0,	0,	0},	/* IDSEL 28 - unused */
